diff --git a/.fern/metadata.json b/.fern/metadata.json
index d8b7bf00e12..6dd605f373f 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -2,6 +2,10 @@
     "cliVersion": "DUMMY",
     "generatorName": "fernapi/fern-typescript-sdk",
     "generatorVersion": "local",
+    "generatorConfig": {
+        "inlinePathParameters": false,
+        "retainOriginalCasing": true
+    },
     "originGitCommit": "DUMMY",
     "originGitCommitIsDirty": null,
     "invokedBy": "ci",
diff --git a/reference.md b/reference.md
index d0792a7198a..aed8aa111ba 100644
--- a/reference.md
+++ b/reference.md
@@ -49,7 +49,7 @@ await client.organizations.getOrganization("organization_id");
 </dl>
 </details>
 
-<details><summary><code>client.organizations.<a href="/src/api/resources/organizations/client/Client.ts">getOrganizationUser</a>({ ...params }) -> SeedPathParameters.User</code></summary>
+<details><summary><code>client.organizations.<a href="/src/api/resources/organizations/client/Client.ts">getOrganizationUser</a>(organization_id, user_id, { ...params }) -> SeedPathParameters.User</code></summary>
 <dl>
 <dd>
 
@@ -62,10 +62,7 @@ await client.organizations.getOrganization("organization_id");
 <dd>
 
 ```typescript
-await client.organizations.getOrganizationUser({
-    organization_id: "organization_id",
-    user_id: "user_id"
-});
+await client.organizations.getOrganizationUser("organization_id", "user_id");
 
 ```
 </dd>
@@ -81,6 +78,22 @@ await client.organizations.getOrganizationUser({
 <dl>
 <dd>
 
+**organization_id:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
+**user_id:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
 **request:** `SeedPathParameters.GetOrganizationUserRequest` 
     
 </dd>
@@ -161,7 +174,7 @@ await client.organizations.searchOrganizations("organization_id", {
 </details>
 
 ## User
-<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">getUser</a>({ ...params }) -> SeedPathParameters.User</code></summary>
+<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">getUser</a>(user_id, { ...params }) -> SeedPathParameters.User</code></summary>
 <dl>
 <dd>
 
@@ -174,9 +187,7 @@ await client.organizations.searchOrganizations("organization_id", {
 <dd>
 
 ```typescript
-await client.user.getUser({
-    user_id: "user_id"
-});
+await client.user.getUser("user_id");
 
 ```
 </dd>
@@ -192,6 +203,14 @@ await client.user.getUser({
 <dl>
 <dd>
 
+**user_id:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
 **request:** `SeedPathParameters.GetUsersRequest` 
     
 </dd>
@@ -264,7 +283,7 @@ await client.user.createUser({
 </dl>
 </details>
 
-<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">updateUser</a>({ ...params }) -> SeedPathParameters.User</code></summary>
+<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">updateUser</a>(user_id, { ...params }) -> SeedPathParameters.User</code></summary>
 <dl>
 <dd>
 
@@ -277,8 +296,7 @@ await client.user.createUser({
 <dd>
 
 ```typescript
-await client.user.updateUser({
-    user_id: "user_id",
+await client.user.updateUser("user_id", {
     body: {
         name: "name",
         tags: ["tags", "tags"]
@@ -299,6 +317,14 @@ await client.user.updateUser({
 <dl>
 <dd>
 
+**user_id:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
 **request:** `SeedPathParameters.UpdateUserRequest` 
     
 </dd>
@@ -319,7 +345,7 @@ await client.user.updateUser({
 </dl>
 </details>
 
-<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">searchUsers</a>({ ...params }) -> SeedPathParameters.User[]</code></summary>
+<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">searchUsers</a>(user_id, { ...params }) -> SeedPathParameters.User[]</code></summary>
 <dl>
 <dd>
 
@@ -332,8 +358,7 @@ await client.user.updateUser({
 <dd>
 
 ```typescript
-await client.user.searchUsers({
-    user_id: "user_id",
+await client.user.searchUsers("user_id", {
     limit: 1
 });
 
@@ -351,6 +376,14 @@ await client.user.searchUsers({
 <dl>
 <dd>
 
+**user_id:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
 **request:** `SeedPathParameters.SearchUsersRequest` 
     
 </dd>
@@ -371,7 +404,7 @@ await client.user.searchUsers({
 </dl>
 </details>
 
-<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">getUserMetadata</a>({ ...params }) -> SeedPathParameters.User</code></summary>
+<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">getUserMetadata</a>(user_id, version, { ...params }) -> SeedPathParameters.User</code></summary>
 <dl>
 <dd>
 
@@ -398,10 +431,7 @@ Test endpoint with path parameter that has a text prefix (v{version})
 <dd>
 
 ```typescript
-await client.user.getUserMetadata({
-    user_id: "user_id",
-    version: 1
-});
+await client.user.getUserMetadata("user_id", 1);
 
 ```
 </dd>
@@ -417,6 +447,22 @@ await client.user.getUserMetadata({
 <dl>
 <dd>
 
+**user_id:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
+**version:** `number` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
 **request:** `SeedPathParameters.GetUserMetadataRequest` 
     
 </dd>
@@ -437,7 +483,7 @@ await client.user.getUserMetadata({
 </dl>
 </details>
 
-<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">getUserSpecifics</a>({ ...params }) -> SeedPathParameters.User</code></summary>
+<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">getUserSpecifics</a>(user_id, version, thought, { ...params }) -> SeedPathParameters.User</code></summary>
 <dl>
 <dd>
 
@@ -464,11 +510,7 @@ Test endpoint with path parameters listed in different order than found in path
 <dd>
 
 ```typescript
-await client.user.getUserSpecifics({
-    user_id: "user_id",
-    version: 1,
-    thought: "thought"
-});
+await client.user.getUserSpecifics("user_id", 1, "thought");
 
 ```
 </dd>
@@ -484,6 +526,30 @@ await client.user.getUserSpecifics({
 <dl>
 <dd>
 
+**user_id:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
+**version:** `number` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
+**thought:** `string` 
+    
+</dd>
+</dl>
+
+<dl>
+<dd>
+
 **request:** `SeedPathParameters.GetUserSpecificsRequest` 
     
 </dd>
diff --git a/snippet.json b/snippet.json
index 6af6e5f9ad4..534a4028803 100644
--- a/snippet.json
+++ b/snippet.json
@@ -19,7 +19,7 @@
             },
             "snippet": {
                 "type": "typescript",
-                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.organizations.getOrganizationUser({\n    organization_id: \"organization_id\",\n    user_id: \"user_id\"\n});\n"
+                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.organizations.getOrganizationUser(\"organization_id\", \"user_id\");\n"
             }
         },
         {
@@ -41,7 +41,7 @@
             },
             "snippet": {
                 "type": "typescript",
-                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.getUser({\n    user_id: \"user_id\"\n});\n"
+                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.getUser(\"user_id\");\n"
             }
         },
         {
@@ -63,7 +63,7 @@
             },
             "snippet": {
                 "type": "typescript",
-                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.updateUser({\n    user_id: \"user_id\",\n    body: {\n        name: \"name\",\n        tags: [\"tags\", \"tags\"]\n    }\n});\n"
+                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.updateUser(\"user_id\", {\n    body: {\n        name: \"name\",\n        tags: [\"tags\", \"tags\"]\n    }\n});\n"
             }
         },
         {
@@ -74,7 +74,7 @@
             },
             "snippet": {
                 "type": "typescript",
-                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.searchUsers({\n    user_id: \"user_id\",\n    limit: 1\n});\n"
+                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.searchUsers(\"user_id\", {\n    limit: 1\n});\n"
             }
         },
         {
@@ -85,7 +85,7 @@
             },
             "snippet": {
                 "type": "typescript",
-                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.getUserMetadata({\n    user_id: \"user_id\",\n    version: 1\n});\n"
+                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.getUserMetadata(\"user_id\", 1);\n"
             }
         },
         {
@@ -96,7 +96,7 @@
             },
             "snippet": {
                 "type": "typescript",
-                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.getUserSpecifics({\n    user_id: \"user_id\",\n    version: 1,\n    thought: \"thought\"\n});\n"
+                "client": "import { SeedPathParametersClient } from \"@fern/path-parameters\";\n\nconst client = new SeedPathParametersClient({ environment: \"YOUR_BASE_URL\" });\nawait client.user.getUserSpecifics(\"user_id\", 1, \"thought\");\n"
             }
         }
     ],
diff --git a/src/api/resources/organizations/client/Client.ts b/src/api/resources/organizations/client/Client.ts
index ffc32534dcb..57abeb76640 100644
--- a/src/api/resources/organizations/client/Client.ts
+++ b/src/api/resources/organizations/client/Client.ts
@@ -76,33 +76,37 @@ export class OrganizationsClient {
     }
 
     /**
+     * @param {string} organization_id
+     * @param {string} user_id
      * @param {SeedPathParameters.GetOrganizationUserRequest} request
      * @param {OrganizationsClient.RequestOptions} requestOptions - Request-specific configuration.
      *
      * @example
-     *     await client.organizations.getOrganizationUser({
-     *         organization_id: "organization_id",
-     *         user_id: "user_id"
-     *     })
+     *     await client.organizations.getOrganizationUser("organization_id", "user_id")
      */
     public getOrganizationUser(
-        request: SeedPathParameters.GetOrganizationUserRequest,
+        organization_id: string,
+        user_id: string,
+        request: SeedPathParameters.GetOrganizationUserRequest = {},
         requestOptions?: OrganizationsClient.RequestOptions,
     ): core.HttpResponsePromise<SeedPathParameters.User> {
-        return core.HttpResponsePromise.fromPromise(this.__getOrganizationUser(request, requestOptions));
+        return core.HttpResponsePromise.fromPromise(
+            this.__getOrganizationUser(organization_id, user_id, request, requestOptions),
+        );
     }
 
     private async __getOrganizationUser(
-        request: SeedPathParameters.GetOrganizationUserRequest,
+        organization_id: string,
+        user_id: string,
+        _request: SeedPathParameters.GetOrganizationUserRequest = {},
         requestOptions?: OrganizationsClient.RequestOptions,
     ): Promise<core.WithRawResponse<SeedPathParameters.User>> {
-        const { organization_id: organizationId, user_id: userId } = request;
         const _headers: core.Fetcher.Args["headers"] = mergeHeaders(this._options?.headers, requestOptions?.headers);
         const _response = await core.fetcher({
             url: core.url.join(
                 (await core.Supplier.get(this._options.baseUrl)) ??
                     (await core.Supplier.get(this._options.environment)),
-                `/${core.url.encodePathParam(this._options.tenant_id)}/organizations/${core.url.encodePathParam(organizationId)}/users/${core.url.encodePathParam(userId)}`,
+                `/${core.url.encodePathParam(this._options.tenant_id)}/organizations/${core.url.encodePathParam(organization_id)}/users/${core.url.encodePathParam(user_id)}`,
             ),
             method: "GET",
             headers: _headers,
diff --git a/src/api/resources/organizations/client/requests/GetOrganizationUserRequest.ts b/src/api/resources/organizations/client/requests/GetOrganizationUserRequest.ts
index 3c2ef11e5f3..25ad763f375 100644
--- a/src/api/resources/organizations/client/requests/GetOrganizationUserRequest.ts
+++ b/src/api/resources/organizations/client/requests/GetOrganizationUserRequest.ts
@@ -2,12 +2,6 @@
 
 /**
  * @example
- *     {
- *         organization_id: "organization_id",
- *         user_id: "user_id"
- *     }
+ *     {}
  */
-export interface GetOrganizationUserRequest {
-    organization_id: string;
-    user_id: string;
-}
+export type GetOrganizationUserRequest = {};
diff --git a/src/api/resources/user/client/Client.ts b/src/api/resources/user/client/Client.ts
index e8ba7f24858..c2137814af5 100644
--- a/src/api/resources/user/client/Client.ts
+++ b/src/api/resources/user/client/Client.ts
@@ -22,32 +22,32 @@ export class UserClient {
     }
 
     /**
+     * @param {string} user_id
      * @param {SeedPathParameters.GetUsersRequest} request
      * @param {UserClient.RequestOptions} requestOptions - Request-specific configuration.
      *
      * @example
-     *     await client.user.getUser({
-     *         user_id: "user_id"
-     *     })
+     *     await client.user.getUser("user_id")
      */
     public getUser(
-        request: SeedPathParameters.GetUsersRequest,
+        user_id: string,
+        request: SeedPathParameters.GetUsersRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): core.HttpResponsePromise<SeedPathParameters.User> {
-        return core.HttpResponsePromise.fromPromise(this.__getUser(request, requestOptions));
+        return core.HttpResponsePromise.fromPromise(this.__getUser(user_id, request, requestOptions));
     }
 
     private async __getUser(
-        request: SeedPathParameters.GetUsersRequest,
+        user_id: string,
+        _request: SeedPathParameters.GetUsersRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): Promise<core.WithRawResponse<SeedPathParameters.User>> {
-        const { user_id: userId } = request;
         const _headers: core.Fetcher.Args["headers"] = mergeHeaders(this._options?.headers, requestOptions?.headers);
         const _response = await core.fetcher({
             url: core.url.join(
                 (await core.Supplier.get(this._options.baseUrl)) ??
                     (await core.Supplier.get(this._options.environment)),
-                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(userId)}`,
+                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(user_id)}`,
             ),
             method: "GET",
             headers: _headers,
@@ -129,12 +129,12 @@ export class UserClient {
     }
 
     /**
+     * @param {string} user_id
      * @param {SeedPathParameters.UpdateUserRequest} request
      * @param {UserClient.RequestOptions} requestOptions - Request-specific configuration.
      *
      * @example
-     *     await client.user.updateUser({
-     *         user_id: "user_id",
+     *     await client.user.updateUser("user_id", {
      *         body: {
      *             name: "name",
      *             tags: ["tags", "tags"]
@@ -142,23 +142,25 @@ export class UserClient {
      *     })
      */
     public updateUser(
+        user_id: string,
         request: SeedPathParameters.UpdateUserRequest,
         requestOptions?: UserClient.RequestOptions,
     ): core.HttpResponsePromise<SeedPathParameters.User> {
-        return core.HttpResponsePromise.fromPromise(this.__updateUser(request, requestOptions));
+        return core.HttpResponsePromise.fromPromise(this.__updateUser(user_id, request, requestOptions));
     }
 
     private async __updateUser(
+        user_id: string,
         request: SeedPathParameters.UpdateUserRequest,
         requestOptions?: UserClient.RequestOptions,
     ): Promise<core.WithRawResponse<SeedPathParameters.User>> {
-        const { user_id: userId, body: _body } = request;
+        const { body: _body } = request;
         const _headers: core.Fetcher.Args["headers"] = mergeHeaders(this._options?.headers, requestOptions?.headers);
         const _response = await core.fetcher({
             url: core.url.join(
                 (await core.Supplier.get(this._options.baseUrl)) ??
                     (await core.Supplier.get(this._options.environment)),
-                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(userId)}`,
+                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(user_id)}`,
             ),
             method: "PATCH",
             headers: _headers,
@@ -188,27 +190,29 @@ export class UserClient {
     }
 
     /**
+     * @param {string} user_id
      * @param {SeedPathParameters.SearchUsersRequest} request
      * @param {UserClient.RequestOptions} requestOptions - Request-specific configuration.
      *
      * @example
-     *     await client.user.searchUsers({
-     *         user_id: "user_id",
+     *     await client.user.searchUsers("user_id", {
      *         limit: 1
      *     })
      */
     public searchUsers(
-        request: SeedPathParameters.SearchUsersRequest,
+        user_id: string,
+        request: SeedPathParameters.SearchUsersRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): core.HttpResponsePromise<SeedPathParameters.User[]> {
-        return core.HttpResponsePromise.fromPromise(this.__searchUsers(request, requestOptions));
+        return core.HttpResponsePromise.fromPromise(this.__searchUsers(user_id, request, requestOptions));
     }
 
     private async __searchUsers(
-        request: SeedPathParameters.SearchUsersRequest,
+        user_id: string,
+        request: SeedPathParameters.SearchUsersRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): Promise<core.WithRawResponse<SeedPathParameters.User[]>> {
-        const { user_id: userId, limit } = request;
+        const { limit } = request;
         const _queryParams: Record<string, unknown> = {
             limit,
         };
@@ -217,7 +221,7 @@ export class UserClient {
             url: core.url.join(
                 (await core.Supplier.get(this._options.baseUrl)) ??
                     (await core.Supplier.get(this._options.environment)),
-                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(userId)}/search`,
+                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(user_id)}/search`,
             ),
             method: "GET",
             headers: _headers,
@@ -256,33 +260,35 @@ export class UserClient {
     /**
      * Test endpoint with path parameter that has a text prefix (v{version})
      *
+     * @param {string} user_id
+     * @param {number} version
      * @param {SeedPathParameters.GetUserMetadataRequest} request
      * @param {UserClient.RequestOptions} requestOptions - Request-specific configuration.
      *
      * @example
-     *     await client.user.getUserMetadata({
-     *         user_id: "user_id",
-     *         version: 1
-     *     })
+     *     await client.user.getUserMetadata("user_id", 1)
      */
     public getUserMetadata(
-        request: SeedPathParameters.GetUserMetadataRequest,
+        user_id: string,
+        version: number,
+        request: SeedPathParameters.GetUserMetadataRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): core.HttpResponsePromise<SeedPathParameters.User> {
-        return core.HttpResponsePromise.fromPromise(this.__getUserMetadata(request, requestOptions));
+        return core.HttpResponsePromise.fromPromise(this.__getUserMetadata(user_id, version, request, requestOptions));
     }
 
     private async __getUserMetadata(
-        request: SeedPathParameters.GetUserMetadataRequest,
+        user_id: string,
+        version: number,
+        _request: SeedPathParameters.GetUserMetadataRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): Promise<core.WithRawResponse<SeedPathParameters.User>> {
-        const { user_id: userId, version } = request;
         const _headers: core.Fetcher.Args["headers"] = mergeHeaders(this._options?.headers, requestOptions?.headers);
         const _response = await core.fetcher({
             url: core.url.join(
                 (await core.Supplier.get(this._options.baseUrl)) ??
                     (await core.Supplier.get(this._options.environment)),
-                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(userId)}/metadata/v${core.url.encodePathParam(version)}`,
+                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(user_id)}/metadata/v${core.url.encodePathParam(version)}`,
             ),
             method: "GET",
             headers: _headers,
@@ -316,34 +322,40 @@ export class UserClient {
     /**
      * Test endpoint with path parameters listed in different order than found in path
      *
+     * @param {string} user_id
+     * @param {number} version
+     * @param {string} thought
      * @param {SeedPathParameters.GetUserSpecificsRequest} request
      * @param {UserClient.RequestOptions} requestOptions - Request-specific configuration.
      *
      * @example
-     *     await client.user.getUserSpecifics({
-     *         user_id: "user_id",
-     *         version: 1,
-     *         thought: "thought"
-     *     })
+     *     await client.user.getUserSpecifics("user_id", 1, "thought")
      */
     public getUserSpecifics(
-        request: SeedPathParameters.GetUserSpecificsRequest,
+        user_id: string,
+        version: number,
+        thought: string,
+        request: SeedPathParameters.GetUserSpecificsRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): core.HttpResponsePromise<SeedPathParameters.User> {
-        return core.HttpResponsePromise.fromPromise(this.__getUserSpecifics(request, requestOptions));
+        return core.HttpResponsePromise.fromPromise(
+            this.__getUserSpecifics(user_id, version, thought, request, requestOptions),
+        );
     }
 
     private async __getUserSpecifics(
-        request: SeedPathParameters.GetUserSpecificsRequest,
+        user_id: string,
+        version: number,
+        thought: string,
+        _request: SeedPathParameters.GetUserSpecificsRequest = {},
         requestOptions?: UserClient.RequestOptions,
     ): Promise<core.WithRawResponse<SeedPathParameters.User>> {
-        const { user_id: userId, version, thought } = request;
         const _headers: core.Fetcher.Args["headers"] = mergeHeaders(this._options?.headers, requestOptions?.headers);
         const _response = await core.fetcher({
             url: core.url.join(
                 (await core.Supplier.get(this._options.baseUrl)) ??
                     (await core.Supplier.get(this._options.environment)),
-                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(userId)}/specifics/${core.url.encodePathParam(version)}/${core.url.encodePathParam(thought)}`,
+                `/${core.url.encodePathParam(this._options.tenant_id)}/user/${core.url.encodePathParam(user_id)}/specifics/${core.url.encodePathParam(version)}/${core.url.encodePathParam(thought)}`,
             ),
             method: "GET",
             headers: _headers,
diff --git a/src/api/resources/user/client/requests/GetUserMetadataRequest.ts b/src/api/resources/user/client/requests/GetUserMetadataRequest.ts
index 8044b0486e4..e0ebbccca1e 100644
--- a/src/api/resources/user/client/requests/GetUserMetadataRequest.ts
+++ b/src/api/resources/user/client/requests/GetUserMetadataRequest.ts
@@ -2,12 +2,6 @@
 
 /**
  * @example
- *     {
- *         user_id: "user_id",
- *         version: 1
- *     }
+ *     {}
  */
-export interface GetUserMetadataRequest {
-    user_id: string;
-    version: number;
-}
+export type GetUserMetadataRequest = {};
diff --git a/src/api/resources/user/client/requests/GetUserSpecificsRequest.ts b/src/api/resources/user/client/requests/GetUserSpecificsRequest.ts
index fd753ec810c..6a04102380c 100644
--- a/src/api/resources/user/client/requests/GetUserSpecificsRequest.ts
+++ b/src/api/resources/user/client/requests/GetUserSpecificsRequest.ts
@@ -2,14 +2,6 @@
 
 /**
  * @example
- *     {
- *         user_id: "user_id",
- *         version: 1,
- *         thought: "thought"
- *     }
+ *     {}
  */
-export interface GetUserSpecificsRequest {
-    user_id: string;
-    version: number;
-    thought: string;
-}
+export type GetUserSpecificsRequest = {};
diff --git a/src/api/resources/user/client/requests/GetUsersRequest.ts b/src/api/resources/user/client/requests/GetUsersRequest.ts
index 860ba8e01a5..04be747552f 100644
--- a/src/api/resources/user/client/requests/GetUsersRequest.ts
+++ b/src/api/resources/user/client/requests/GetUsersRequest.ts
@@ -2,10 +2,6 @@
 
 /**
  * @example
- *     {
- *         user_id: "user_id"
- *     }
+ *     {}
  */
-export interface GetUsersRequest {
-    user_id: string;
-}
+export type GetUsersRequest = {};
diff --git a/src/api/resources/user/client/requests/SearchUsersRequest.ts b/src/api/resources/user/client/requests/SearchUsersRequest.ts
index 86cfd1efae9..5709da2747a 100644
--- a/src/api/resources/user/client/requests/SearchUsersRequest.ts
+++ b/src/api/resources/user/client/requests/SearchUsersRequest.ts
@@ -3,11 +3,9 @@
 /**
  * @example
  *     {
- *         user_id: "user_id",
  *         limit: 1
  *     }
  */
 export interface SearchUsersRequest {
-    user_id: string;
     limit?: number;
 }
diff --git a/src/api/resources/user/client/requests/UpdateUserRequest.ts b/src/api/resources/user/client/requests/UpdateUserRequest.ts
index a114d09121b..f9e46a83a2b 100644
--- a/src/api/resources/user/client/requests/UpdateUserRequest.ts
+++ b/src/api/resources/user/client/requests/UpdateUserRequest.ts
@@ -5,7 +5,6 @@ import type * as SeedPathParameters from "../../../../index.js";
 /**
  * @example
  *     {
- *         user_id: "user_id",
  *         body: {
  *             name: "name",
  *             tags: ["tags", "tags"]
@@ -13,6 +12,5 @@ import type * as SeedPathParameters from "../../../../index.js";
  *     }
  */
 export interface UpdateUserRequest {
-    user_id: string;
     body: SeedPathParameters.User;
 }
diff --git a/tests/wire/organizations.test.ts b/tests/wire/organizations.test.ts
index 0724ddb7965..babee74479f 100644
--- a/tests/wire/organizations.test.ts
+++ b/tests/wire/organizations.test.ts
@@ -44,10 +44,7 @@ describe("OrganizationsClient", () => {
             .jsonBody(rawResponseBody)
             .build();
 
-        const response = await client.organizations.getOrganizationUser({
-            organization_id: "organization_id",
-            user_id: "user_id",
-        });
+        const response = await client.organizations.getOrganizationUser("organization_id", "user_id");
         expect(response).toEqual(rawResponseBody);
     });
 
diff --git a/tests/wire/user.test.ts b/tests/wire/user.test.ts
index afad4e65b72..bae1536660d 100644
--- a/tests/wire/user.test.ts
+++ b/tests/wire/user.test.ts
@@ -22,9 +22,7 @@ describe("UserClient", () => {
             .jsonBody(rawResponseBody)
             .build();
 
-        const response = await client.user.getUser({
-            user_id: "user_id",
-        });
+        const response = await client.user.getUser("user_id");
         expect(response).toEqual(rawResponseBody);
     });
 
@@ -73,8 +71,7 @@ describe("UserClient", () => {
             .jsonBody(rawResponseBody)
             .build();
 
-        const response = await client.user.updateUser({
-            user_id: "user_id",
+        const response = await client.user.updateUser("user_id", {
             body: {
                 name: "name",
                 tags: ["tags", "tags"],
@@ -104,8 +101,7 @@ describe("UserClient", () => {
             .jsonBody(rawResponseBody)
             .build();
 
-        const response = await client.user.searchUsers({
-            user_id: "user_id",
+        const response = await client.user.searchUsers("user_id", {
             limit: 1,
         });
         expect(response).toEqual(rawResponseBody);
@@ -129,10 +125,7 @@ describe("UserClient", () => {
             .jsonBody(rawResponseBody)
             .build();
 
-        const response = await client.user.getUserMetadata({
-            user_id: "user_id",
-            version: 1,
-        });
+        const response = await client.user.getUserMetadata("user_id", 1);
         expect(response).toEqual(rawResponseBody);
     });
 
@@ -154,11 +147,7 @@ describe("UserClient", () => {
             .jsonBody(rawResponseBody)
             .build();
 
-        const response = await client.user.getUserSpecifics({
-            user_id: "user_id",
-            version: 1,
-            thought: "thought",
-        });
+        const response = await client.user.getUserSpecifics("user_id", 1, "thought");
         expect(response).toEqual(rawResponseBody);
     });
 });
