diff --git a/.fern/metadata.json b/.fern/metadata.json
index 2ecd82f71ab..63a2062b5d1 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -2,6 +2,9 @@
   "cliVersion": "DUMMY",
   "generatorName": "fernapi/fern-python-sdk",
   "generatorVersion": "local",
+  "generatorConfig": {
+    "use_typeddict_requests": true
+  },
   "originGitCommit": "DUMMY",
   "originGitCommitIsDirty": null,
   "invokedBy": "ci",
diff --git a/snippet.json b/snippet.json
index 9b543bf50d5..0f4376ac8fb 100644
--- a/snippet.json
+++ b/snippet.json
@@ -22,8 +22,8 @@
                 "identifier_override": "endpoint_nullable.createUser"
             },
             "snippet": {
-                "sync_client": "import datetime\n\nfrom seed import SeedNullable\nfrom seed.nullable import Metadata, Status\n\nclient = SeedNullable(\n    base_url=\"https://yourhost.com/path/to/api\",\n)\nclient.nullable.create_user(\n    username=\"username\",\n    tags=[\"tags\", \"tags\"],\n    metadata=Metadata(\n        created_at=datetime.datetime.fromisoformat(\n            \"2024-01-15 09:30:00+00:00\",\n        ),\n        updated_at=datetime.datetime.fromisoformat(\n            \"2024-01-15 09:30:00+00:00\",\n        ),\n        avatar=\"avatar\",\n        activated=True,\n        status=Status(),\n        values={\"values\": \"values\"},\n    ),\n    avatar=\"avatar\",\n)\n",
-                "async_client": "import asyncio\nimport datetime\n\nfrom seed import AsyncSeedNullable\nfrom seed.nullable import Metadata, Status\n\nclient = AsyncSeedNullable(\n    base_url=\"https://yourhost.com/path/to/api\",\n)\n\n\nasync def main() -> None:\n    await client.nullable.create_user(\n        username=\"username\",\n        tags=[\"tags\", \"tags\"],\n        metadata=Metadata(\n            created_at=datetime.datetime.fromisoformat(\n                \"2024-01-15 09:30:00+00:00\",\n            ),\n            updated_at=datetime.datetime.fromisoformat(\n                \"2024-01-15 09:30:00+00:00\",\n            ),\n            avatar=\"avatar\",\n            activated=True,\n            status=Status(),\n            values={\"values\": \"values\"},\n        ),\n        avatar=\"avatar\",\n    )\n\n\nasyncio.run(main())\n",
+                "sync_client": "import datetime\n\nfrom seed import SeedNullable\n\nclient = SeedNullable(\n    base_url=\"https://yourhost.com/path/to/api\",\n)\nclient.nullable.create_user(\n    username=\"username\",\n    tags=[\"tags\", \"tags\"],\n    metadata={\n        \"created_at\": datetime.datetime.fromisoformat(\n            \"2024-01-15 09:30:00+00:00\",\n        ),\n        \"updated_at\": datetime.datetime.fromisoformat(\n            \"2024-01-15 09:30:00+00:00\",\n        ),\n        \"avatar\": \"avatar\",\n        \"activated\": True,\n        \"status\": {\"type\": \"active\"},\n        \"values\": {\"values\": \"values\"},\n    },\n    avatar=\"avatar\",\n)\n",
+                "async_client": "import asyncio\nimport datetime\n\nfrom seed import AsyncSeedNullable\n\nclient = AsyncSeedNullable(\n    base_url=\"https://yourhost.com/path/to/api\",\n)\n\n\nasync def main() -> None:\n    await client.nullable.create_user(\n        username=\"username\",\n        tags=[\"tags\", \"tags\"],\n        metadata={\n            \"created_at\": datetime.datetime.fromisoformat(\n                \"2024-01-15 09:30:00+00:00\",\n            ),\n            \"updated_at\": datetime.datetime.fromisoformat(\n                \"2024-01-15 09:30:00+00:00\",\n            ),\n            \"avatar\": \"avatar\",\n            \"activated\": True,\n            \"status\": {\"type\": \"active\"},\n            \"values\": {\"values\": \"values\"},\n        },\n        avatar=\"avatar\",\n    )\n\n\nasyncio.run(main())\n",
                 "type": "python"
             }
         },
diff --git a/src/seed/__init__.py b/src/seed/__init__.py
index bd71717429e..74967b3b4e3 100644
--- a/src/seed/__init__.py
+++ b/src/seed/__init__.py
@@ -12,13 +12,20 @@ if typing.TYPE_CHECKING:
     from .nullable import (
         Email,
         Metadata,
+        MetadataParams,
         Status,
+        StatusParams,
         Status_Active,
+        Status_ActiveParams,
         Status_Archived,
+        Status_ArchivedParams,
         Status_SoftDeleted,
+        Status_SoftDeletedParams,
         User,
         UserId,
+        UserParams,
         WeirdNumber,
+        WeirdNumberParams,
     )
     from .version import __version__
 _dynamic_imports: typing.Dict[str, str] = {
@@ -27,14 +34,21 @@ _dynamic_imports: typing.Dict[str, str] = {
     "DefaultAsyncHttpxClient": "._default_clients",
     "Email": ".nullable",
     "Metadata": ".nullable",
+    "MetadataParams": ".nullable",
     "SeedNullable": ".client",
     "Status": ".nullable",
+    "StatusParams": ".nullable",
     "Status_Active": ".nullable",
+    "Status_ActiveParams": ".nullable",
     "Status_Archived": ".nullable",
+    "Status_ArchivedParams": ".nullable",
     "Status_SoftDeleted": ".nullable",
+    "Status_SoftDeletedParams": ".nullable",
     "User": ".nullable",
     "UserId": ".nullable",
+    "UserParams": ".nullable",
     "WeirdNumber": ".nullable",
+    "WeirdNumberParams": ".nullable",
     "__version__": ".version",
     "nullable": ".nullable",
 }
@@ -67,14 +81,21 @@ __all__ = [
     "DefaultAsyncHttpxClient",
     "Email",
     "Metadata",
+    "MetadataParams",
     "SeedNullable",
     "Status",
+    "StatusParams",
     "Status_Active",
+    "Status_ActiveParams",
     "Status_Archived",
+    "Status_ArchivedParams",
     "Status_SoftDeleted",
+    "Status_SoftDeletedParams",
     "User",
     "UserId",
+    "UserParams",
     "WeirdNumber",
+    "WeirdNumberParams",
     "__version__",
     "nullable",
 ]
diff --git a/src/seed/nullable/__init__.py b/src/seed/nullable/__init__.py
index 2d0783e8d92..b9351e6144a 100644
--- a/src/seed/nullable/__init__.py
+++ b/src/seed/nullable/__init__.py
@@ -17,16 +17,32 @@ if typing.TYPE_CHECKING:
         UserId,
         WeirdNumber,
     )
+    from .requests import (
+        MetadataParams,
+        StatusParams,
+        Status_ActiveParams,
+        Status_ArchivedParams,
+        Status_SoftDeletedParams,
+        UserParams,
+        WeirdNumberParams,
+    )
 _dynamic_imports: typing.Dict[str, str] = {
     "Email": ".types",
     "Metadata": ".types",
+    "MetadataParams": ".requests",
     "Status": ".types",
+    "StatusParams": ".requests",
     "Status_Active": ".types",
+    "Status_ActiveParams": ".requests",
     "Status_Archived": ".types",
+    "Status_ArchivedParams": ".requests",
     "Status_SoftDeleted": ".types",
+    "Status_SoftDeletedParams": ".requests",
     "User": ".types",
     "UserId": ".types",
+    "UserParams": ".requests",
     "WeirdNumber": ".types",
+    "WeirdNumberParams": ".requests",
 }
 
 
@@ -54,11 +70,18 @@ def __dir__():
 __all__ = [
     "Email",
     "Metadata",
+    "MetadataParams",
     "Status",
+    "StatusParams",
     "Status_Active",
+    "Status_ActiveParams",
     "Status_Archived",
+    "Status_ArchivedParams",
     "Status_SoftDeleted",
+    "Status_SoftDeletedParams",
     "User",
     "UserId",
+    "UserParams",
     "WeirdNumber",
+    "WeirdNumberParams",
 ]
diff --git a/src/seed/nullable/client.py b/src/seed/nullable/client.py
index dfa7a184aaf..069cb3d0c31 100644
--- a/src/seed/nullable/client.py
+++ b/src/seed/nullable/client.py
@@ -5,7 +5,7 @@ import typing
 from ..core.client_wrapper import AsyncClientWrapper, SyncClientWrapper
 from ..core.request_options import RequestOptions
 from .raw_client import AsyncRawNullableClient, RawNullableClient
-from .types.metadata import Metadata
+from .requests.metadata import MetadataParams
 from .types.user import User
 
 # this is used as the default value for optional parameters
@@ -87,7 +87,7 @@ class NullableClient:
         *,
         username: str,
         tags: typing.Optional[typing.Sequence[str]] = OMIT,
-        metadata: typing.Optional[Metadata] = OMIT,
+        metadata: typing.Optional[MetadataParams] = OMIT,
         avatar: typing.Optional[str] = OMIT,
         request_options: typing.Optional[RequestOptions] = None,
     ) -> User:
@@ -98,7 +98,7 @@ class NullableClient:
 
         tags : typing.Optional[typing.Sequence[str]]
 
-        metadata : typing.Optional[Metadata]
+        metadata : typing.Optional[MetadataParams]
 
         avatar : typing.Optional[str]
 
@@ -114,7 +114,6 @@ class NullableClient:
         import datetime
 
         from seed import SeedNullable
-        from seed.nullable import Metadata, Status
 
         client = SeedNullable(
             base_url="https://yourhost.com/path/to/api",
@@ -122,18 +121,18 @@ class NullableClient:
         client.nullable.create_user(
             username="username",
             tags=["tags", "tags"],
-            metadata=Metadata(
-                created_at=datetime.datetime.fromisoformat(
+            metadata={
+                "created_at": datetime.datetime.fromisoformat(
                     "2024-01-15 09:30:00+00:00",
                 ),
-                updated_at=datetime.datetime.fromisoformat(
+                "updated_at": datetime.datetime.fromisoformat(
                     "2024-01-15 09:30:00+00:00",
                 ),
-                avatar="avatar",
-                activated=True,
-                status=Status(),
-                values={"values": "values"},
-            ),
+                "avatar": "avatar",
+                "activated": True,
+                "status": {"type": "active"},
+                "values": {"values": "values"},
+            },
             avatar="avatar",
         )
         """
@@ -256,7 +255,7 @@ class AsyncNullableClient:
         *,
         username: str,
         tags: typing.Optional[typing.Sequence[str]] = OMIT,
-        metadata: typing.Optional[Metadata] = OMIT,
+        metadata: typing.Optional[MetadataParams] = OMIT,
         avatar: typing.Optional[str] = OMIT,
         request_options: typing.Optional[RequestOptions] = None,
     ) -> User:
@@ -267,7 +266,7 @@ class AsyncNullableClient:
 
         tags : typing.Optional[typing.Sequence[str]]
 
-        metadata : typing.Optional[Metadata]
+        metadata : typing.Optional[MetadataParams]
 
         avatar : typing.Optional[str]
 
@@ -284,7 +283,6 @@ class AsyncNullableClient:
         import datetime
 
         from seed import AsyncSeedNullable
-        from seed.nullable import Metadata, Status
 
         client = AsyncSeedNullable(
             base_url="https://yourhost.com/path/to/api",
@@ -295,18 +293,18 @@ class AsyncNullableClient:
             await client.nullable.create_user(
                 username="username",
                 tags=["tags", "tags"],
-                metadata=Metadata(
-                    created_at=datetime.datetime.fromisoformat(
+                metadata={
+                    "created_at": datetime.datetime.fromisoformat(
                         "2024-01-15 09:30:00+00:00",
                     ),
-                    updated_at=datetime.datetime.fromisoformat(
+                    "updated_at": datetime.datetime.fromisoformat(
                         "2024-01-15 09:30:00+00:00",
                     ),
-                    avatar="avatar",
-                    activated=True,
-                    status=Status(),
-                    values={"values": "values"},
-                ),
+                    "avatar": "avatar",
+                    "activated": True,
+                    "status": {"type": "active"},
+                    "values": {"values": "values"},
+                },
                 avatar="avatar",
             )
 
diff --git a/src/seed/nullable/raw_client.py b/src/seed/nullable/raw_client.py
index eccee6f090d..7c25be50351 100644
--- a/src/seed/nullable/raw_client.py
+++ b/src/seed/nullable/raw_client.py
@@ -10,7 +10,7 @@ from ..core.parse_error import ParsingError
 from ..core.pydantic_utilities import parse_obj_as
 from ..core.request_options import RequestOptions
 from ..core.serialization import convert_and_respect_annotation_metadata
-from .types.metadata import Metadata
+from .requests.metadata import MetadataParams
 from .types.user import User
 from pydantic import ValidationError
 
@@ -88,7 +88,7 @@ class RawNullableClient:
         *,
         username: str,
         tags: typing.Optional[typing.Sequence[str]] = OMIT,
-        metadata: typing.Optional[Metadata] = OMIT,
+        metadata: typing.Optional[MetadataParams] = OMIT,
         avatar: typing.Optional[str] = OMIT,
         request_options: typing.Optional[RequestOptions] = None,
     ) -> HttpResponse[User]:
@@ -99,7 +99,7 @@ class RawNullableClient:
 
         tags : typing.Optional[typing.Sequence[str]]
 
-        metadata : typing.Optional[Metadata]
+        metadata : typing.Optional[MetadataParams]
 
         avatar : typing.Optional[str]
 
@@ -117,7 +117,7 @@ class RawNullableClient:
                 "username": username,
                 "tags": tags,
                 "metadata": convert_and_respect_annotation_metadata(
-                    object_=metadata, annotation=Metadata, direction="write"
+                    object_=metadata, annotation=MetadataParams, direction="write"
                 ),
                 "avatar": avatar,
             },
@@ -258,7 +258,7 @@ class AsyncRawNullableClient:
         *,
         username: str,
         tags: typing.Optional[typing.Sequence[str]] = OMIT,
-        metadata: typing.Optional[Metadata] = OMIT,
+        metadata: typing.Optional[MetadataParams] = OMIT,
         avatar: typing.Optional[str] = OMIT,
         request_options: typing.Optional[RequestOptions] = None,
     ) -> AsyncHttpResponse[User]:
@@ -269,7 +269,7 @@ class AsyncRawNullableClient:
 
         tags : typing.Optional[typing.Sequence[str]]
 
-        metadata : typing.Optional[Metadata]
+        metadata : typing.Optional[MetadataParams]
 
         avatar : typing.Optional[str]
 
@@ -287,7 +287,7 @@ class AsyncRawNullableClient:
                 "username": username,
                 "tags": tags,
                 "metadata": convert_and_respect_annotation_metadata(
-                    object_=metadata, annotation=Metadata, direction="write"
+                    object_=metadata, annotation=MetadataParams, direction="write"
                 ),
                 "avatar": avatar,
             },
diff --git a/src/seed/nullable/requests/__init__.py b/src/seed/nullable/requests/__init__.py
new file mode 100644
index 00000000000..1113c999758
--- /dev/null
+++ b/src/seed/nullable/requests/__init__.py
@@ -0,0 +1,53 @@
+# This file was auto-generated by Fern from our API Definition.
+
+# isort: skip_file
+
+import typing
+from importlib import import_module
+
+if typing.TYPE_CHECKING:
+    from .metadata import MetadataParams
+    from .status import StatusParams, Status_ActiveParams, Status_ArchivedParams, Status_SoftDeletedParams
+    from .user import UserParams
+    from .weird_number import WeirdNumberParams
+_dynamic_imports: typing.Dict[str, str] = {
+    "MetadataParams": ".metadata",
+    "StatusParams": ".status",
+    "Status_ActiveParams": ".status",
+    "Status_ArchivedParams": ".status",
+    "Status_SoftDeletedParams": ".status",
+    "UserParams": ".user",
+    "WeirdNumberParams": ".weird_number",
+}
+
+
+def __getattr__(attr_name: str) -> typing.Any:
+    module_name = _dynamic_imports.get(attr_name)
+    if module_name is None:
+        raise AttributeError(f"No {attr_name} found in _dynamic_imports for module name -> {__name__}")
+    try:
+        module = import_module(module_name, __package__)
+        if module_name == f".{attr_name}":
+            return module
+        else:
+            return getattr(module, attr_name)
+    except ImportError as e:
+        raise ImportError(f"Failed to import {attr_name} from {module_name}: {e}") from e
+    except AttributeError as e:
+        raise AttributeError(f"Failed to get {attr_name} from {module_name}: {e}") from e
+
+
+def __dir__():
+    lazy_attrs = list(_dynamic_imports.keys())
+    return sorted(lazy_attrs)
+
+
+__all__ = [
+    "MetadataParams",
+    "StatusParams",
+    "Status_ActiveParams",
+    "Status_ArchivedParams",
+    "Status_SoftDeletedParams",
+    "UserParams",
+    "WeirdNumberParams",
+]
diff --git a/src/seed/nullable/requests/metadata.py b/src/seed/nullable/requests/metadata.py
new file mode 100644
index 00000000000..055fa5a9389
--- /dev/null
+++ b/src/seed/nullable/requests/metadata.py
@@ -0,0 +1,17 @@
+# This file was auto-generated by Fern from our API Definition.
+
+import datetime as dt
+import typing
+
+import typing_extensions
+from ...core.serialization import FieldMetadata
+from .status import StatusParams
+
+
+class MetadataParams(typing_extensions.TypedDict):
+    created_at: typing_extensions.Annotated[dt.datetime, FieldMetadata(alias="createdAt")]
+    updated_at: typing_extensions.Annotated[dt.datetime, FieldMetadata(alias="updatedAt")]
+    avatar: typing.Optional[str]
+    activated: typing_extensions.NotRequired[typing.Optional[bool]]
+    status: StatusParams
+    values: typing_extensions.NotRequired[typing.Dict[str, typing.Optional[str]]]
diff --git a/src/seed/nullable/requests/status.py b/src/seed/nullable/requests/status.py
new file mode 100644
index 00000000000..56f563a0b0c
--- /dev/null
+++ b/src/seed/nullable/requests/status.py
@@ -0,0 +1,25 @@
+# This file was auto-generated by Fern from our API Definition.
+
+from __future__ import annotations
+
+import datetime as dt
+import typing
+
+import typing_extensions
+
+
+class Status_ActiveParams(typing_extensions.TypedDict):
+    type: typing.Literal["active"]
+
+
+class Status_ArchivedParams(typing_extensions.TypedDict):
+    value: typing.Optional[dt.datetime]
+    type: typing.Literal["archived"]
+
+
+class Status_SoftDeletedParams(typing_extensions.TypedDict):
+    value: typing_extensions.NotRequired[dt.datetime]
+    type: typing.Literal["soft-deleted"]
+
+
+StatusParams = typing.Union[Status_ActiveParams, Status_ArchivedParams, Status_SoftDeletedParams]
diff --git a/src/seed/nullable/requests/user.py b/src/seed/nullable/requests/user.py
new file mode 100644
index 00000000000..3f777d7a235
--- /dev/null
+++ b/src/seed/nullable/requests/user.py
@@ -0,0 +1,21 @@
+# This file was auto-generated by Fern from our API Definition.
+
+import typing
+
+import typing_extensions
+from ...core.serialization import FieldMetadata
+from ..types.email import Email
+from ..types.user_id import UserId
+from .metadata import MetadataParams
+from .weird_number import WeirdNumberParams
+
+
+class UserParams(typing_extensions.TypedDict):
+    name: str
+    id: UserId
+    tags: typing.Optional[typing.Sequence[str]]
+    metadata: typing_extensions.NotRequired[typing.Optional[MetadataParams]]
+    email: Email
+    favorite_number: typing_extensions.Annotated[WeirdNumberParams, FieldMetadata(alias="favorite-number")]
+    numbers: typing_extensions.NotRequired[typing.Optional[typing.Sequence[int]]]
+    strings: typing_extensions.NotRequired[typing.Optional[typing.Dict[str, typing.Any]]]
diff --git a/src/seed/nullable/requests/weird_number.py b/src/seed/nullable/requests/weird_number.py
new file mode 100644
index 00000000000..a4185267492
--- /dev/null
+++ b/src/seed/nullable/requests/weird_number.py
@@ -0,0 +1,5 @@
+# This file was auto-generated by Fern from our API Definition.
+
+import typing
+
+WeirdNumberParams = typing.Union[int, typing.Optional[float], typing.Optional[str], float]
