diff --git a/dist/cjs/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.ts b/dist/cjs/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.ts
index 6e0cfbf323afb6ac9286a5333aa21c33b2de216f..f0844727502daf61d30dc915f6bb0d983633010a 100644
--- a/dist/cjs/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.ts
+++ b/dist/cjs/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.ts
@@ -10,5 +10,6 @@ export interface AsymmetricKeySignatureVerification {
      */
     signaturePrefix: string | undefined;
     keySource: FernIr.AsymmetricKeySource;
+    payloadFormat: FernIr.WebhookPayloadFormat | undefined;
     timestamp: FernIr.WebhookTimestampConfig | undefined;
 }
diff --git a/dist/esm/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.mts b/dist/esm/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.mts
index 0b5ec1c63f50797fe2cb5f0cc4150627b4f3904e..790a7ceb8f4aa0ab8f3a15f156985a6c12fbf023 100644
--- a/dist/esm/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.mts
+++ b/dist/esm/api/resources/webhooks/types/AsymmetricKeySignatureVerification.d.mts
@@ -10,5 +10,6 @@ export interface AsymmetricKeySignatureVerification {
      */
     signaturePrefix: string | undefined;
     keySource: FernIr.AsymmetricKeySource;
+    payloadFormat: FernIr.WebhookPayloadFormat | undefined;
     timestamp: FernIr.WebhookTimestampConfig | undefined;
 }
