diff --git a/.fern/metadata.json b/.fern/metadata.json
index 8d01069edb2..69d9e30ad07 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -2,6 +2,9 @@
     "cliVersion": "DUMMY",
     "generatorName": "fernapi/fern-typescript-sdk",
     "generatorVersion": "local",
+    "generatorConfig": {
+        "customPagerName": "MyPager"
+    },
     "originGitCommit": "DUMMY",
     "originGitCommitIsDirty": null,
     "invokedBy": "manual",
diff --git a/reference.md b/reference.md
index e3a26b9cd8c..63175e58671 100644
--- a/reference.md
+++ b/reference.md
@@ -1,6 +1,6 @@
 # Reference
 ## Users
-<details><summary><code>client.users.<a href="/src/api/resources/users/client/Client.ts">listWithCustomPager</a>({ ...params }) -> core.CustomPager&lt;string, SeedPagination.UsersListResponse&gt;</code></summary>
+<details><summary><code>client.users.<a href="/src/api/resources/users/client/Client.ts">listWithCustomPager</a>({ ...params }) -> core.MyPager&lt;string, SeedPagination.UsersListResponse&gt;</code></summary>
 <dl>
 <dd>
 
diff --git a/src/api/resources/users/client/Client.ts b/src/api/resources/users/client/Client.ts
index 38eed1096c7..c506db8c2fa 100644
--- a/src/api/resources/users/client/Client.ts
+++ b/src/api/resources/users/client/Client.ts
@@ -34,7 +34,7 @@ export class UsersClient {
     public async listWithCustomPager(
         request: SeedPagination.ListWithCustomPagerRequest = {},
         requestOptions?: UsersClient.RequestOptions,
-    ): Promise<core.CustomPager<string, SeedPagination.UsersListResponse>> {
+    ): Promise<core.MyPager<string, SeedPagination.UsersListResponse>> {
         const { limit, starting_after: startingAfter } = request;
         const _queryParams: Record<string, unknown> = {
             limit,
@@ -75,7 +75,7 @@ export class UsersClient {
             }
             return handleNonStatusCodeError(_response.error, _response.rawResponse, "GET", "/users");
         };
-        return core.createCustomPager<string, SeedPagination.UsersListResponse>({
+        return core.createMyPager<string, SeedPagination.UsersListResponse>({
             sendRequest: _sendRequest,
             initialHttpRequest: _request,
             clientOptions: this._options,
diff --git a/src/core/index.ts b/src/core/index.ts
index 948a9d6d9e2..108dddd88ae 100644
--- a/src/core/index.ts
+++ b/src/core/index.ts
@@ -2,6 +2,6 @@ export * from "./auth/index.js";
 export * from "./base64.js";
 export * from "./fetcher/index.js";
 export * as logging from "./logging/index.js";
-export { type CustomPager, createCustomPager } from "./pagination/CustomPager.js";
+export { createMyPager, type MyPager } from "./pagination/MyPager.js";
 export * from "./runtime/index.js";
 export * as url from "./url/index.js";
diff --git a/src/core/pagination/CustomPager.ts b/src/core/pagination/MyPager.ts
similarity index 96%
rename from /Users/Patrick/fern/fern2/seed/ts-sdk/pagination-custom/baseline/src/core/pagination/CustomPager.ts
rename to /private/var/folders/27/s5yzs3z956ndrqw_kpcj1yb00000gq/T/tmp-71563-De1SGME0orQV/src/core/pagination/MyPager.ts
index f9bb9186552..6d4bdd2d7b2 100644
--- a/src/core/pagination/CustomPager.ts
+++ b/src/core/pagination/MyPager.ts
@@ -8,7 +8,7 @@ import type { RawResponse } from "../fetcher/index.js";
  * @template TItem The type of the items in the page.
  * @template TResponse The type of the API response.
  */
-export class CustomPager<TItem, TResponse> implements AsyncIterable<TItem> {
+export class MyPager<TItem, TResponse> implements AsyncIterable<TItem> {
     /** The items from the current page */
     public data: TItem[];
     /** The raw HTTP response */
@@ -142,7 +142,7 @@ export class CustomPager<TItem, TResponse> implements AsyncIterable<TItem> {
     }
 }
 
-export async function createCustomPager<TItem, TResponse>({
+export async function createMyPager<TItem, TResponse>({
     sendRequest,
     initialHttpRequest,
     clientOptions,
@@ -151,7 +151,7 @@ export async function createCustomPager<TItem, TResponse>({
     initialHttpRequest: Fetcher.Args;
     clientOptions: NormalizedClientOptions;
     requestOptions?: BaseRequestOptions;
-}): Promise<CustomPager<TItem, TResponse>> {
+}): Promise<MyPager<TItem, TResponse>> {
     const response = await sendRequest(initialHttpRequest);
     if (!response.ok) {
         const reason =
@@ -161,7 +161,7 @@ export async function createCustomPager<TItem, TResponse>({
     const data = response.body;
     const rawResponse = response.rawResponse;
     const parsed = await parse<TItem, TResponse>({ request: initialHttpRequest, data, rawResponse });
-    return new CustomPager({
+    return new MyPager({
         response: data,
         rawResponse,
         items: parsed.items,
