diff --git a/.fern/metadata.json b/.fern/metadata.json
index 6533addd8b4..13a85e33522 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -3,7 +3,8 @@
   "generatorName": "fernapi/fern-ruby-sdk-v2",
   "generatorVersion": "local",
   "generatorConfig": {
-    "rubocopSeverity": "error"
+    "rubocopSeverity": "error",
+    "offsetSemantics": "page-index"
   },
   "originGitCommit": "DUMMY",
   "originGitCommitIsDirty": null,
diff --git a/lib/seed/inline_users/inline_users/client.rb b/lib/seed/inline_users/inline_users/client.rb
index 1aa54ca288f..8b9bf9087af 100644
--- a/lib/seed/inline_users/inline_users/client.rb
+++ b/lib/seed/inline_users/inline_users/client.rb
@@ -327,7 +327,7 @@ module Seed
             initial_page: query_params["page"],
             item_field: :users,
             has_next_field: nil,
-            step: true
+            step: false
           ) do |next_page|
             query_params["page"] = next_page
             request = Seed::Internal::JSON::Request.new(
@@ -379,7 +379,7 @@ module Seed
             initial_page: query_params["page"],
             item_field: :users,
             has_next_field: :has_next_page,
-            step: true
+            step: false
           ) do |next_page|
             query_params["page"] = next_page
             request = Seed::Internal::JSON::Request.new(
diff --git a/lib/seed/users/client.rb b/lib/seed/users/client.rb
index 62c007e9cd7..7ecf65285c3 100644
--- a/lib/seed/users/client.rb
+++ b/lib/seed/users/client.rb
@@ -371,7 +371,7 @@ module Seed
           initial_page: query_params["page"],
           item_field: :data,
           has_next_field: nil,
-          step: true
+          step: false
         ) do |next_page|
           query_params["page"] = next_page
           request = Seed::Internal::JSON::Request.new(
@@ -423,7 +423,7 @@ module Seed
           initial_page: query_params["page"],
           item_field: :data,
           has_next_field: :has_next_page,
-          step: true
+          step: false
         ) do |next_page|
           query_params["page"] = next_page
           request = Seed::Internal::JSON::Request.new(
