diff --git a/.fern/metadata.json b/.fern/metadata.json
index 7bcb89e7c5c..1b02f5f60a3 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -3,7 +3,8 @@
   "generatorName": "fernapi/fern-ruby-sdk-v2",
   "generatorVersion": "local",
   "generatorConfig": {
-    "rubocopSeverity": "error"
+    "rubocopSeverity": "error",
+    "omitFernHeaders": true
   },
   "originGitCommit": "DUMMY",
   "invokedBy": "manual",
diff --git a/lib/seed/client.rb b/lib/seed/client.rb
index e9059bafddd..bb1d0c97d6e 100644
--- a/lib/seed/client.rb
+++ b/lib/seed/client.rb
@@ -72,8 +72,6 @@ module Seed
       @raw_client = Seed::Internal::Http::RawClient.new(
         base_url: base_url,
         headers: {
-          "User-Agent" => "fern_examples/0.0.1",
-          "X-Fern-Language" => "Ruby",
           Authorization: "Bearer #{token}"
         }
       )
diff --git a/lib/seed/internal/http/raw_client.rb b/lib/seed/internal/http/raw_client.rb
index f4f53404497..324db115136 100644
--- a/lib/seed/internal/http/raw_client.rb
+++ b/lib/seed/internal/http/raw_client.rb
@@ -25,11 +25,7 @@ module Seed
           @base_url = base_url
           @max_retries = max_retries
           @timeout = timeout
-          @default_headers = {
-            "X-Fern-Language": "Ruby",
-            "X-Fern-SDK-Name": "seed",
-            "X-Fern-SDK-Version": "0.0.1"
-          }.merge(headers)
+          @default_headers = headers
         end
 
         # @param request [Seed::Internal::Http::BaseRequest] The HTTP request.
