diff --git a/.fern/metadata.json b/.fern/metadata.json
index e206b6bea6d..16cde5c1cd4 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -3,7 +3,11 @@
   "generatorName": "fernapi/fern-go-sdk",
   "generatorVersion": "local",
   "generatorConfig": {
-    "enableWireTests": false
+    "enableWireTests": false,
+    "packageName": "upload",
+    "module": {
+      "path": "github.com/fern-api/file-upload-go"
+    }
   },
   "originGitCommit": "DUMMY",
   "originGitCommitIsDirty": null,
diff --git a/README.md b/README.md
index 71dde44cb31..d504fcf3f60 100644
--- a/README.md
+++ b/README.md
@@ -33,13 +33,13 @@ import (
     context "context"
     strings "strings"
 
-    fern "github.com/file-upload/fern"
-    client "github.com/file-upload/fern/client"
+    upload "github.com/fern-api/file-upload-go"
+    client "github.com/fern-api/file-upload-go/client"
 )
 
 func do() {
     client := client.NewClient()
-    request := &fern.JustFileRequest{
+    request := &upload.JustFileRequest{
         File: strings.NewReader(
             "",
         ),
diff --git a/client/client.go b/client/client.go
index 77957d91ee8..40f31a2c940 100644
--- a/client/client.go
+++ b/client/client.go
@@ -3,10 +3,10 @@
 package client
 
 import (
-	core "github.com/file-upload/fern/core"
-	internal "github.com/file-upload/fern/internal"
-	option "github.com/file-upload/fern/option"
-	service "github.com/file-upload/fern/service"
+	core "github.com/fern-api/file-upload-go/core"
+	internal "github.com/fern-api/file-upload-go/internal"
+	option "github.com/fern-api/file-upload-go/option"
+	service "github.com/fern-api/file-upload-go/service"
 )
 
 type Client struct {
diff --git a/client/client_test.go b/client/client_test.go
index cc4a6db32e6..a59ea570052 100644
--- a/client/client_test.go
+++ b/client/client_test.go
@@ -3,7 +3,7 @@
 package client
 
 import (
-	option "github.com/file-upload/fern/option"
+	option "github.com/fern-api/file-upload-go/option"
 	assert "github.com/stretchr/testify/assert"
 	http "net/http"
 	testing "testing"
diff --git a/core/request_option.go b/core/request_option.go
index 28fea1c2512..60c817be3e0 100644
--- a/core/request_option.go
+++ b/core/request_option.go
@@ -49,7 +49,7 @@ func (r *RequestOptions) ToHeader() http.Header { return r.cloneHeader() }
 func (r *RequestOptions) cloneHeader() http.Header {
 	headers := r.HTTPHeader.Clone()
 	headers.Set("X-Fern-Language", "Go")
-	headers.Set("X-Fern-SDK-Name", "github.com/file-upload/fern")
+	headers.Set("X-Fern-SDK-Name", "github.com/fern-api/file-upload-go")
 	headers.Set("X-Fern-SDK-Version", "v0.0.1")
 	headers.Set("User-Agent", "github.com/file-upload/fern/0.0.1")
 	return headers
diff --git a/dynamic-snippets/example0/snippet.go b/dynamic-snippets/example0/snippet.go
index 3a2b2c37707..5edfff080e6 100644
--- a/dynamic-snippets/example0/snippet.go
+++ b/dynamic-snippets/example0/snippet.go
@@ -4,9 +4,9 @@ import (
     context "context"
     strings "strings"
 
-    fern "github.com/file-upload/fern"
-    client "github.com/file-upload/fern/client"
-    option "github.com/file-upload/fern/option"
+    upload "github.com/fern-api/file-upload-go"
+    client "github.com/fern-api/file-upload-go/client"
+    option "github.com/fern-api/file-upload-go/option"
 )
 
 func do() {
@@ -15,7 +15,7 @@ func do() {
             "https://api.fern.com",
         ),
     )
-    request := &fern.JustFileRequest{
+    request := &upload.JustFileRequest{
         File: strings.NewReader(
             "",
         ),
diff --git a/dynamic-snippets/example1/snippet.go b/dynamic-snippets/example1/snippet.go
index 58dcab120b0..366fdffc246 100644
--- a/dynamic-snippets/example1/snippet.go
+++ b/dynamic-snippets/example1/snippet.go
@@ -4,9 +4,9 @@ import (
     context "context"
     strings "strings"
 
-    fern "github.com/file-upload/fern"
-    client "github.com/file-upload/fern/client"
-    option "github.com/file-upload/fern/option"
+    upload "github.com/fern-api/file-upload-go"
+    client "github.com/fern-api/file-upload-go/client"
+    option "github.com/fern-api/file-upload-go/option"
 )
 
 func do() {
@@ -15,7 +15,7 @@ func do() {
             "https://api.fern.com",
         ),
     )
-    request := &fern.OptionalArgsRequest{
+    request := &upload.OptionalArgsRequest{
         ImageFile: strings.NewReader(
             "",
         ),
diff --git a/dynamic-snippets/example2/snippet.go b/dynamic-snippets/example2/snippet.go
index 0b986d680db..5d1ce74581a 100644
--- a/dynamic-snippets/example2/snippet.go
+++ b/dynamic-snippets/example2/snippet.go
@@ -3,8 +3,8 @@ package example
 import (
     context "context"
 
-    client "github.com/file-upload/fern/client"
-    option "github.com/file-upload/fern/option"
+    client "github.com/fern-api/file-upload-go/client"
+    option "github.com/fern-api/file-upload-go/option"
 )
 
 func do() {
diff --git a/error_codes.go b/error_codes.go
index fe894f78c6d..bb9f7aebb8b 100644
--- a/error_codes.go
+++ b/error_codes.go
@@ -1,9 +1,9 @@
 // Code generated by Fern. DO NOT EDIT.
 
-package fileupload
+package upload
 
 import (
-	internal "github.com/file-upload/fern/internal"
+	internal "github.com/fern-api/file-upload-go/internal"
 )
 
 var ErrorCodes internal.ErrorCodes = internal.ErrorCodes{}
diff --git a/file_param.go b/file_param.go
index d8d1190319f..85a210d9292 100644
--- a/file_param.go
+++ b/file_param.go
@@ -1,4 +1,4 @@
-package fileupload
+package upload
 
 import (
 	"io"
diff --git a/go.mod b/go.mod
index ebf53790cd9..42a0e0626d6 100644
--- a/go.mod
+++ b/go.mod
@@ -1,4 +1,4 @@
-module github.com/file-upload/fern
+module github.com/fern-api/file-upload-go
 
 go 1.21
 
diff --git a/internal/caller.go b/internal/caller.go
index 82c42be1aa6..3a84e08790a 100644
--- a/internal/caller.go
+++ b/internal/caller.go
@@ -12,7 +12,7 @@ import (
 	"reflect"
 	"strings"
 
-	"github.com/file-upload/fern/core"
+	"github.com/fern-api/file-upload-go/core"
 )
 
 const (
diff --git a/internal/caller_test.go b/internal/caller_test.go
index 321122ffef8..27fc95f4838 100644
--- a/internal/caller_test.go
+++ b/internal/caller_test.go
@@ -14,7 +14,7 @@ import (
 	"strings"
 	"testing"
 
-	"github.com/file-upload/fern/core"
+	"github.com/fern-api/file-upload-go/core"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
diff --git a/internal/error_decoder.go b/internal/error_decoder.go
index 6d26e7e5ccb..337791bce41 100644
--- a/internal/error_decoder.go
+++ b/internal/error_decoder.go
@@ -8,7 +8,7 @@ import (
 	"io"
 	"net/http"
 
-	"github.com/file-upload/fern/core"
+	"github.com/fern-api/file-upload-go/core"
 )
 
 // ErrorCodes maps HTTP status codes to error constructors.
diff --git a/internal/error_decoder_test.go b/internal/error_decoder_test.go
index 1ac4b561114..92c5a4cc3d9 100644
--- a/internal/error_decoder_test.go
+++ b/internal/error_decoder_test.go
@@ -6,7 +6,7 @@ import (
 	"net/http"
 	"testing"
 
-	"github.com/file-upload/fern/core"
+	"github.com/fern-api/file-upload-go/core"
 	"github.com/stretchr/testify/assert"
 )
 
diff --git a/internal/retrier_test.go b/internal/retrier_test.go
index 695325d8149..adcc5b7a6c0 100644
--- a/internal/retrier_test.go
+++ b/internal/retrier_test.go
@@ -10,7 +10,7 @@ import (
 	"testing"
 	"time"
 
-	"github.com/file-upload/fern/core"
+	"github.com/fern-api/file-upload-go/core"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
diff --git a/option/request_option.go b/option/request_option.go
index 772d004c479..15c614cc4de 100644
--- a/option/request_option.go
+++ b/option/request_option.go
@@ -3,7 +3,7 @@
 package option
 
 import (
-	core "github.com/file-upload/fern/core"
+	core "github.com/fern-api/file-upload-go/core"
 	http "net/http"
 	url "net/url"
 )
diff --git a/pointer.go b/pointer.go
index cb67cd5c767..09819160e1b 100644
--- a/pointer.go
+++ b/pointer.go
@@ -1,4 +1,4 @@
-package fileupload
+package upload
 
 import (
 	"time"
diff --git a/pointer_test.go b/pointer_test.go
index 1b3eaf698d4..d00d4e5a670 100644
--- a/pointer_test.go
+++ b/pointer_test.go
@@ -1,4 +1,4 @@
-package fileupload
+package upload
 
 import (
 	"testing"
diff --git a/reference.md b/reference.md
index d2ccc9525c3..c7754729c06 100644
--- a/reference.md
+++ b/reference.md
@@ -13,7 +13,7 @@
 <dd>
 
 ```go
-request := &fern.JustFileRequest{
+request := &upload.JustFileRequest{
         File: strings.NewReader(
             "",
         ),
@@ -47,7 +47,7 @@ client.Service.JustFile(
 <dd>
 
 ```go
-request := &fern.OptionalArgsRequest{
+request := &upload.OptionalArgsRequest{
         ImageFile: strings.NewReader(
             "",
         ),
diff --git a/service/client.go b/service/client.go
index 11db7537f9a..3f8001ee848 100644
--- a/service/client.go
+++ b/service/client.go
@@ -5,10 +5,10 @@ package service
 import (
 	context "context"
 
-	fern "github.com/file-upload/fern"
-	core "github.com/file-upload/fern/core"
-	internal "github.com/file-upload/fern/internal"
-	option "github.com/file-upload/fern/option"
+	upload "github.com/fern-api/file-upload-go"
+	core "github.com/fern-api/file-upload-go/core"
+	internal "github.com/fern-api/file-upload-go/internal"
+	option "github.com/fern-api/file-upload-go/option"
 )
 
 type Client struct {
@@ -35,7 +35,7 @@ func NewClient(options *core.RequestOptions) *Client {
 
 func (c *Client) Post(
 	ctx context.Context,
-	request *fern.MyRequest,
+	request *upload.MyRequest,
 	opts ...option.RequestOption,
 ) error {
 	_, err := c.WithRawResponse.Post(
@@ -51,7 +51,7 @@ func (c *Client) Post(
 
 func (c *Client) JustFile(
 	ctx context.Context,
-	request *fern.JustFileRequest,
+	request *upload.JustFileRequest,
 	opts ...option.RequestOption,
 ) error {
 	_, err := c.WithRawResponse.JustFile(
@@ -67,7 +67,7 @@ func (c *Client) JustFile(
 
 func (c *Client) JustFileWithQueryParams(
 	ctx context.Context,
-	request *fern.JustFileWithQueryParamsRequest,
+	request *upload.JustFileWithQueryParamsRequest,
 	opts ...option.RequestOption,
 ) error {
 	_, err := c.WithRawResponse.JustFileWithQueryParams(
@@ -83,7 +83,7 @@ func (c *Client) JustFileWithQueryParams(
 
 func (c *Client) JustFileWithOptionalQueryParams(
 	ctx context.Context,
-	request *fern.JustFileWithOptionalQueryParamsRequest,
+	request *upload.JustFileWithOptionalQueryParamsRequest,
 	opts ...option.RequestOption,
 ) error {
 	_, err := c.WithRawResponse.JustFileWithOptionalQueryParams(
@@ -99,7 +99,7 @@ func (c *Client) JustFileWithOptionalQueryParams(
 
 func (c *Client) WithContentType(
 	ctx context.Context,
-	request *fern.WithContentTypeRequest,
+	request *upload.WithContentTypeRequest,
 	opts ...option.RequestOption,
 ) error {
 	_, err := c.WithRawResponse.WithContentType(
@@ -115,7 +115,7 @@ func (c *Client) WithContentType(
 
 func (c *Client) WithFormEncoding(
 	ctx context.Context,
-	request *fern.WithFormEncodingRequest,
+	request *upload.WithFormEncodingRequest,
 	opts ...option.RequestOption,
 ) error {
 	_, err := c.WithRawResponse.WithFormEncoding(
@@ -131,7 +131,7 @@ func (c *Client) WithFormEncoding(
 
 func (c *Client) WithFormEncodedContainers(
 	ctx context.Context,
-	request *fern.MyOtherRequest,
+	request *upload.MyOtherRequest,
 	opts ...option.RequestOption,
 ) error {
 	_, err := c.WithRawResponse.WithFormEncodedContainers(
@@ -147,7 +147,7 @@ func (c *Client) WithFormEncodedContainers(
 
 func (c *Client) OptionalArgs(
 	ctx context.Context,
-	request *fern.OptionalArgsRequest,
+	request *upload.OptionalArgsRequest,
 	opts ...option.RequestOption,
 ) (string, error) {
 	response, err := c.WithRawResponse.OptionalArgs(
@@ -163,7 +163,7 @@ func (c *Client) OptionalArgs(
 
 func (c *Client) WithInlineType(
 	ctx context.Context,
-	request *fern.InlineTypeRequest,
+	request *upload.InlineTypeRequest,
 	opts ...option.RequestOption,
 ) (string, error) {
 	response, err := c.WithRawResponse.WithInlineType(
@@ -179,7 +179,7 @@ func (c *Client) WithInlineType(
 
 func (c *Client) WithJSONProperty(
 	ctx context.Context,
-	request *fern.WithJSONPropertyRequest,
+	request *upload.WithJSONPropertyRequest,
 	opts ...option.RequestOption,
 ) (string, error) {
 	response, err := c.WithRawResponse.WithJSONProperty(
@@ -209,7 +209,7 @@ func (c *Client) Simple(
 
 func (c *Client) WithLiteralAndEnumTypes(
 	ctx context.Context,
-	request *fern.LiteralEnumRequest,
+	request *upload.LiteralEnumRequest,
 	opts ...option.RequestOption,
 ) (string, error) {
 	response, err := c.WithRawResponse.WithLiteralAndEnumTypes(
diff --git a/service/raw_client.go b/service/raw_client.go
index ceb7ee663f1..36c030e755d 100644
--- a/service/raw_client.go
+++ b/service/raw_client.go
@@ -7,10 +7,10 @@ import (
 	fmt "fmt"
 	http "net/http"
 
-	fern "github.com/file-upload/fern"
-	core "github.com/file-upload/fern/core"
-	internal "github.com/file-upload/fern/internal"
-	option "github.com/file-upload/fern/option"
+	upload "github.com/fern-api/file-upload-go"
+	core "github.com/fern-api/file-upload-go/core"
+	internal "github.com/fern-api/file-upload-go/internal"
+	option "github.com/fern-api/file-upload-go/option"
 )
 
 type RawClient struct {
@@ -34,7 +34,7 @@ func NewRawClient(options *core.RequestOptions) *RawClient {
 
 func (r *RawClient) Post(
 	ctx context.Context,
-	request *fern.MyRequest,
+	request *upload.MyRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[any], error) {
 	options := core.NewRequestOptions(opts...)
@@ -148,7 +148,7 @@ func (r *RawClient) Post(
 
 func (r *RawClient) JustFile(
 	ctx context.Context,
-	request *fern.JustFileRequest,
+	request *upload.JustFileRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[any], error) {
 	options := core.NewRequestOptions(opts...)
@@ -196,7 +196,7 @@ func (r *RawClient) JustFile(
 
 func (r *RawClient) JustFileWithQueryParams(
 	ctx context.Context,
-	request *fern.JustFileWithQueryParamsRequest,
+	request *upload.JustFileWithQueryParamsRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[any], error) {
 	options := core.NewRequestOptions(opts...)
@@ -251,7 +251,7 @@ func (r *RawClient) JustFileWithQueryParams(
 
 func (r *RawClient) JustFileWithOptionalQueryParams(
 	ctx context.Context,
-	request *fern.JustFileWithOptionalQueryParamsRequest,
+	request *upload.JustFileWithOptionalQueryParamsRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[any], error) {
 	options := core.NewRequestOptions(opts...)
@@ -306,7 +306,7 @@ func (r *RawClient) JustFileWithOptionalQueryParams(
 
 func (r *RawClient) WithContentType(
 	ctx context.Context,
-	request *fern.WithContentTypeRequest,
+	request *upload.WithContentTypeRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[any], error) {
 	options := core.NewRequestOptions(opts...)
@@ -365,7 +365,7 @@ func (r *RawClient) WithContentType(
 
 func (r *RawClient) WithFormEncoding(
 	ctx context.Context,
-	request *fern.WithFormEncodingRequest,
+	request *upload.WithFormEncodingRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[any], error) {
 	options := core.NewRequestOptions(opts...)
@@ -419,7 +419,7 @@ func (r *RawClient) WithFormEncoding(
 
 func (r *RawClient) WithFormEncodedContainers(
 	ctx context.Context,
-	request *fern.MyOtherRequest,
+	request *upload.MyOtherRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[any], error) {
 	options := core.NewRequestOptions(opts...)
@@ -538,7 +538,7 @@ func (r *RawClient) WithFormEncodedContainers(
 
 func (r *RawClient) OptionalArgs(
 	ctx context.Context,
-	request *fern.OptionalArgsRequest,
+	request *upload.OptionalArgsRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[string], error) {
 	options := core.NewRequestOptions(opts...)
@@ -595,7 +595,7 @@ func (r *RawClient) OptionalArgs(
 
 func (r *RawClient) WithInlineType(
 	ctx context.Context,
-	request *fern.InlineTypeRequest,
+	request *upload.InlineTypeRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[string], error) {
 	options := core.NewRequestOptions(opts...)
@@ -648,7 +648,7 @@ func (r *RawClient) WithInlineType(
 
 func (r *RawClient) WithJSONProperty(
 	ctx context.Context,
-	request *fern.WithJSONPropertyRequest,
+	request *upload.WithJSONPropertyRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[string], error) {
 	options := core.NewRequestOptions(opts...)
@@ -740,7 +740,7 @@ func (r *RawClient) Simple(
 
 func (r *RawClient) WithLiteralAndEnumTypes(
 	ctx context.Context,
-	request *fern.LiteralEnumRequest,
+	request *upload.LiteralEnumRequest,
 	opts ...option.RequestOption,
 ) (*core.Response[string], error) {
 	options := core.NewRequestOptions(opts...)
diff --git a/service.go b/service.go
index d194a185ffc..e6866c8ad7e 100644
--- a/service.go
+++ b/service.go
@@ -1,11 +1,11 @@
 // Code generated by Fern. DO NOT EDIT.
 
-package fileupload
+package upload
 
 import (
 	json "encoding/json"
 	fmt "fmt"
-	internal "github.com/file-upload/fern/internal"
+	internal "github.com/fern-api/file-upload-go/internal"
 	io "io"
 	big "math/big"
 )
diff --git a/service_test.go b/service_test.go
index 4c0d4f38b5a..d26a0e02da9 100644
--- a/service_test.go
+++ b/service_test.go
@@ -1,6 +1,6 @@
 // Code generated by Fern. DO NOT EDIT.
 
-package fileupload
+package upload
 
 import (
 	json "encoding/json"
diff --git a/snippet.json b/snippet.json
index 1ba8f33720f..897919e8afd 100644
--- a/snippet.json
+++ b/snippet.json
@@ -8,7 +8,7 @@
             },
             "snippet": {
                 "type": "go",
-                "client": "import (\n\tcontext \"context\"\n\tfern \"github.com/file-upload/fern\"\n\tfernclient \"github.com/file-upload/fern/client\"\n)\n\nclient := fernclient.NewClient()\nerr := client.Service.JustFile(\n\tcontext.TODO(),\n\t\u0026fern.JustFileRequest{},\n)\n"
+                "client": "import (\n\tcontext \"context\"\n\tfileuploadgo \"github.com/fern-api/file-upload-go\"\n\tfileuploadgoclient \"github.com/fern-api/file-upload-go/client\"\n)\n\nclient := fileuploadgoclient.NewClient()\nerr := client.Service.JustFile(\n\tcontext.TODO(),\n\t\u0026fileuploadgo.JustFileRequest{},\n)\n"
             }
         },
         {
@@ -19,7 +19,7 @@
             },
             "snippet": {
                 "type": "go",
-                "client": "import (\n\tcontext \"context\"\n\tfern \"github.com/file-upload/fern\"\n\tfernclient \"github.com/file-upload/fern/client\"\n)\n\nclient := fernclient.NewClient()\nresponse, err := client.Service.OptionalArgs(\n\tcontext.TODO(),\n\t\u0026fern.OptionalArgsRequest{},\n)\n"
+                "client": "import (\n\tcontext \"context\"\n\tfileuploadgo \"github.com/fern-api/file-upload-go\"\n\tfileuploadgoclient \"github.com/fern-api/file-upload-go/client\"\n)\n\nclient := fileuploadgoclient.NewClient()\nresponse, err := client.Service.OptionalArgs(\n\tcontext.TODO(),\n\t\u0026fileuploadgo.OptionalArgsRequest{},\n)\n"
             }
         },
         {
@@ -30,7 +30,7 @@
             },
             "snippet": {
                 "type": "go",
-                "client": "import (\n\tcontext \"context\"\n\tfernclient \"github.com/file-upload/fern/client\"\n)\n\nclient := fernclient.NewClient()\nerr := client.Service.Simple(\n\tcontext.TODO(),\n)\n"
+                "client": "import (\n\tcontext \"context\"\n\tfileuploadgoclient \"github.com/fern-api/file-upload-go/client\"\n)\n\nclient := fileuploadgoclient.NewClient()\nerr := client.Service.Simple(\n\tcontext.TODO(),\n)\n"
             }
         }
     ]
