diff --git a/.fern/metadata.json b/.fern/metadata.json
index 5e4188a44b1..e0368df5f3a 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -2,7 +2,9 @@
   "cliVersion": "DUMMY",
   "generatorName": "fernapi/fern-csharp-sdk",
   "generatorVersion": "local",
-  "generatorConfig": {},
+  "generatorConfig": {
+    "unified-client-options": true
+  },
   "originGitCommit": "DUMMY",
   "originGitCommitIsDirty": null,
   "invokedBy": "ci",
diff --git a/README.md b/README.md
index 228e8ac9be9..4a4e02ea9de 100644
--- a/README.md
+++ b/README.md
@@ -41,7 +41,9 @@ Instantiate and use the client with the following:
 ```csharp
 using SeedBearerTokenEnvironmentVariable;
 
-var client = new SeedBearerTokenEnvironmentVariableClient("API_KEY");
+var client = new SeedBearerTokenEnvironmentVariableClient(
+    clientOptions: new ClientOptions { ApiKey = "API_KEY" }
+);
 await client.Service.GetWithBearerTokenAsync();
 ```
 
diff --git a/Snippets/Example0.cs b/Snippets/Example0.cs
index ded604adb11..a60b23af66a 100644
--- a/Snippets/Example0.cs
+++ b/Snippets/Example0.cs
@@ -4,8 +4,8 @@ public partial class Examples
 {
     public async Task Example0() {
         var client = new SeedBearerTokenEnvironmentVariableClient(
-            apiKey: "<token>",
             clientOptions: new ClientOptions {
+                ApiKey = "<token>",
                 BaseUrl = "https://api.fern.com"
             }
         );
diff --git a/snippet.json b/snippet.json
index fa19c0f21f0..37e8575575d 100644
--- a/snippet.json
+++ b/snippet.json
@@ -10,7 +10,7 @@
             },
             "snippet": {
                 "type": "csharp",
-                "client": "using SeedBearerTokenEnvironmentVariable;\n\nvar client = new SeedBearerTokenEnvironmentVariableClient(\"API_KEY\");\nawait client.Service.GetWithBearerTokenAsync();\n"
+                "client": "using SeedBearerTokenEnvironmentVariable;\n\nvar client = new SeedBearerTokenEnvironmentVariableClient(\n    clientOptions: new ClientOptions { ApiKey = \"API_KEY\" }\n);\nawait client.Service.GetWithBearerTokenAsync();\n"
             }
         }
     ]
diff --git a/src/SeedBearerTokenEnvironmentVariable/Core/Public/ClientOptions.cs b/src/SeedBearerTokenEnvironmentVariable/Core/Public/ClientOptions.cs
index 718b2dacd6a..fc847679675 100644
--- a/src/SeedBearerTokenEnvironmentVariable/Core/Public/ClientOptions.cs
+++ b/src/SeedBearerTokenEnvironmentVariable/Core/Public/ClientOptions.cs
@@ -1,3 +1,4 @@
+using global::System.Diagnostics.CodeAnalysis;
 using SeedBearerTokenEnvironmentVariable.Core;
 
 namespace SeedBearerTokenEnvironmentVariable;
@@ -5,6 +6,20 @@ namespace SeedBearerTokenEnvironmentVariable;
 [Serializable]
 public partial class ClientOptions
 {
+    public ClientOptions() { }
+
+    [SetsRequiredMembers]
+    internal ClientOptions(ClientOptions other)
+    {
+        BaseUrl = other.BaseUrl;
+        HttpClient = other.HttpClient;
+        MaxRetries = other.MaxRetries;
+        Timeout = other.Timeout;
+        Headers = new Headers(new Dictionary<string, HeaderValue>(other.Headers));
+        AdditionalHeaders = other.AdditionalHeaders;
+        ApiKey = other.ApiKey;
+    }
+
     /// <summary>
     /// The http headers sent with the request.
     /// </summary>
@@ -13,13 +28,13 @@ public partial class ClientOptions
     /// <summary>
     /// The Base URL for the API.
     /// </summary>
-    public string BaseUrl { get;
+    public required string BaseUrl { get;
 #if NET5_0_OR_GREATER
         init;
 #else
         set;
 #endif
-    } = "";
+    }
 
     /// <summary>
     /// The http client used to make requests.
@@ -74,19 +89,16 @@ public partial class ClientOptions
 #endif
     } = null;
 
+    /// <summary>
+    /// The apiKey to use for authentication.
+    /// </summary>
+    public string? ApiKey { get; set; }
+
     /// <summary>
     /// Clones this and returns a new instance
     /// </summary>
     internal ClientOptions Clone()
     {
-        return new ClientOptions
-        {
-            BaseUrl = BaseUrl,
-            HttpClient = HttpClient,
-            MaxRetries = MaxRetries,
-            Timeout = Timeout,
-            Headers = new Headers(new Dictionary<string, HeaderValue>(Headers)),
-            AdditionalHeaders = AdditionalHeaders,
-        };
+        return new ClientOptions(this);
     }
 }
diff --git a/src/SeedBearerTokenEnvironmentVariable/SeedBearerTokenEnvironmentVariableClient.cs b/src/SeedBearerTokenEnvironmentVariable/SeedBearerTokenEnvironmentVariableClient.cs
index 4c161008b1f..ec2b3aeb765 100644
--- a/src/SeedBearerTokenEnvironmentVariable/SeedBearerTokenEnvironmentVariableClient.cs
+++ b/src/SeedBearerTokenEnvironmentVariable/SeedBearerTokenEnvironmentVariableClient.cs
@@ -7,16 +7,12 @@ public partial class SeedBearerTokenEnvironmentVariableClient
 {
     private readonly RawClient _client;
 
-    public SeedBearerTokenEnvironmentVariableClient(
-        string? apiKey = null,
-        ClientOptions? clientOptions = null
-    )
+    public SeedBearerTokenEnvironmentVariableClient(ClientOptions clientOptions)
     {
-        apiKey ??= GetFromEnvironmentOrThrow(
+        clientOptions.ApiKey ??= GetFromEnvironmentOrThrow(
             "COURIER_API_KEY",
             "Please pass in apiKey or set the environment variable COURIER_API_KEY."
         );
-        clientOptions ??= new ClientOptions();
         var platformHeaders = new Headers(
             new Dictionary<string, string>()
             {
@@ -41,7 +37,7 @@ public partial class SeedBearerTokenEnvironmentVariableClient
         var authHeaders = new Headers(
             new Dictionary<string, string>()
             {
-                { "Authorization", $"Bearer {apiKey ?? ""}" },
+                { "Authorization", $"Bearer {clientOptions.ApiKey ?? ""}" },
                 { "X-API-Version", "1.0.0" },
             }
         );
diff --git a/src/SeedBearerTokenEnvironmentVariable.Test/Core/RawClientTests/RetriesTests.cs b/src/SeedBearerTokenEnvironmentVariable.Test/Core/RawClientTests/RetriesTests.cs
index 7edf27e4786..ec47dd78972 100644
--- a/src/SeedBearerTokenEnvironmentVariable.Test/Core/RawClientTests/RetriesTests.cs
+++ b/src/SeedBearerTokenEnvironmentVariable.Test/Core/RawClientTests/RetriesTests.cs
@@ -26,7 +26,12 @@ public class RetriesTests
         _baseUrl = _server.Url ?? "";
         _httpClient = new HttpClient { BaseAddress = new Uri(_baseUrl) };
         _rawClient = new RawClient(
-            new ClientOptions { HttpClient = _httpClient, MaxRetries = MaxRetries }
+            new ClientOptions
+            {
+                HttpClient = _httpClient,
+                MaxRetries = MaxRetries,
+                BaseUrl = "http://localhost",
+            }
         )
         {
             BaseRetryDelay = 0,
diff --git a/src/SeedBearerTokenEnvironmentVariable.Test/Unit/MockServer/BaseMockServerTest.cs b/src/SeedBearerTokenEnvironmentVariable.Test/Unit/MockServer/BaseMockServerTest.cs
index 8d4ecdaa115..0a031fb0422 100644
--- a/src/SeedBearerTokenEnvironmentVariable.Test/Unit/MockServer/BaseMockServerTest.cs
+++ b/src/SeedBearerTokenEnvironmentVariable.Test/Unit/MockServer/BaseMockServerTest.cs
@@ -24,8 +24,12 @@ public class BaseMockServerTest
 
         // Initialize the Client
         Client = new SeedBearerTokenEnvironmentVariableClient(
-            "API_KEY",
-            clientOptions: new ClientOptions { BaseUrl = Server.Urls[0], MaxRetries = 0 }
+            clientOptions: new ClientOptions
+            {
+                ApiKey = "API_KEY",
+                BaseUrl = Server.Urls[0],
+                MaxRetries = 0,
+            }
         );
     }
 
