diff --git a/.fern/metadata.json b/.fern/metadata.json
index d8b7bf00e12..77a2b5c4285 100644
--- a/.fern/metadata.json
+++ b/.fern/metadata.json
@@ -2,6 +2,17 @@
     "cliVersion": "DUMMY",
     "generatorName": "fernapi/fern-typescript-sdk",
     "generatorVersion": "local",
+    "generatorConfig": {
+        "naming": {
+            "namespace": "acme",
+            "client": "AcmeSdkClient",
+            "error": "AcmeApiError",
+            "timeoutError": "AcmeRequestTimeoutError",
+            "environment": "AcmeApiEnvironment",
+            "environmentUrls": "AcmeApiEnvironmentUrls",
+            "version": "AcmeApiVersion"
+        }
+    },
     "originGitCommit": "DUMMY",
     "originGitCommitIsDirty": null,
     "invokedBy": "ci",
diff --git a/README.md b/README.md
index e5f3d14b95b..8fb20f7e2cc 100644
--- a/README.md
+++ b/README.md
@@ -40,9 +40,9 @@ A full reference for this library is available [here](./reference.md).
 Instantiate and use the client with the following:
 
 ```typescript
-import { SeedSimpleApiClient, SeedSimpleApiEnvironment } from "@fern/simple-api";
+import { AcmeApiEnvironment, AcmeSdkClient } from "@fern/simple-api";
 
-const client = new SeedSimpleApiClient({ environment: SeedSimpleApiEnvironment.Production, token: "YOUR_TOKEN" });
+const client = new AcmeSdkClient({ environment: AcmeApiEnvironment.Production, token: "YOUR_TOKEN" });
 await client.user.get("id");
 ```
 
@@ -51,10 +51,10 @@ await client.user.get("id");
 This SDK allows you to configure different environments for API requests.
 
 ```typescript
-import { SeedSimpleApiClient, SeedSimpleApiEnvironment } from "@fern/simple-api";
+import { AcmeSdkClient, acmeEnvironment } from "@fern/simple-api";
 
-const client = new SeedSimpleApiClient({
-    environment: SeedSimpleApiEnvironment.Production,
+const client = new AcmeSdkClient({
+    environment: acmeEnvironment.Production,
 });
 ```
 
@@ -64,12 +64,12 @@ When the API returns a non-success status code (4xx or 5xx response), a subclass
 will be thrown.
 
 ```typescript
-import { SeedSimpleApiError } from "@fern/simple-api";
+import { AcmeApiError } from "@fern/simple-api";
 
 try {
     await client.user.get(...);
 } catch (err) {
-    if (err instanceof SeedSimpleApiError) {
+    if (err instanceof AcmeApiError) {
         console.log(err.statusCode);
         console.log(err.message);
         console.log(err.body);
@@ -95,9 +95,9 @@ const client = new UserClient({...});
 If you would like to send additional headers as part of the request, use the `headers` request option.
 
 ```typescript
-import { SeedSimpleApiClient } from "@fern/simple-api";
+import { AcmeSdkClient } from "@fern/simple-api";
 
-const client = new SeedSimpleApiClient({
+const client = new AcmeSdkClient({
     ...
     headers: {
         'X-Custom-Header': 'custom value'
@@ -182,9 +182,9 @@ console.log(rawResponse.headers['X-My-Header']);
 The SDK supports logging. You can configure the logger by passing in a `logging` object to the client options.
 
 ```typescript
-import { SeedSimpleApiClient, logging } from "@fern/simple-api";
+import { AcmeSdkClient, logging } from "@fern/simple-api";
 
-const client = new SeedSimpleApiClient({
+const client = new AcmeSdkClient({
     ...
     logging: {
         level: logging.LogLevel.Debug, // defaults to logging.LogLevel.Info
diff --git a/reference.md b/reference.md
index d73128c61a3..a980ad5d8a0 100644
--- a/reference.md
+++ b/reference.md
@@ -1,6 +1,6 @@
 # Reference
 ## User
-<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">get</a>(id) -> SeedSimpleApi.User</code></summary>
+<details><summary><code>client.user.<a href="/src/api/resources/user/client/Client.ts">get</a>(id) -> acme.User</code></summary>
 <dl>
 <dd>
 
diff --git a/snippet.json b/snippet.json
index a66ba5b7e79..2c6d04e2f0c 100644
--- a/snippet.json
+++ b/snippet.json
@@ -8,7 +8,7 @@
             },
             "snippet": {
                 "type": "typescript",
-                "client": "import { SeedSimpleApiClient, SeedSimpleApiEnvironment } from \"@fern/simple-api\";\n\nconst client = new SeedSimpleApiClient({ environment: SeedSimpleApiEnvironment.Production, token: \"YOUR_TOKEN\" });\nawait client.user.get(\"id\");\n"
+                "client": "import { AcmeApiEnvironment, AcmeSdkClient } from \"@fern/simple-api\";\n\nconst client = new AcmeSdkClient({ environment: AcmeApiEnvironment.Production, token: \"YOUR_TOKEN\" });\nawait client.user.get(\"id\");\n"
             }
         }
     ],
diff --git a/src/BaseClient.ts b/src/BaseClient.ts
index 339461e0a49..7b6db7b27d3 100644
--- a/src/BaseClient.ts
+++ b/src/BaseClient.ts
@@ -6,7 +6,7 @@ import * as core from "./core/index.js";
 import type * as environments from "./environments.js";
 
 export type BaseClientOptions = {
-    environment: core.Supplier<environments.SeedSimpleApiEnvironment | string>;
+    environment: core.Supplier<environments.AcmeApiEnvironment | string>;
     /** Specify a custom URL to connect the client to. */
     baseUrl?: core.Supplier<string>;
     /** Additional headers to include in requests. */
diff --git a/src/Client.ts b/src/Client.ts
index 9ed57372443..227ef03a913 100644
--- a/src/Client.ts
+++ b/src/Client.ts
@@ -5,17 +5,17 @@ import type { BaseClientOptions, BaseRequestOptions } from "./BaseClient.js";
 import { type NormalizedClientOptionsWithAuth, normalizeClientOptionsWithAuth } from "./BaseClient.js";
 import * as core from "./core/index.js";
 
-export declare namespace SeedSimpleApiClient {
+export declare namespace AcmeSdkClient {
     export type Options = BaseClientOptions;
 
     export interface RequestOptions extends BaseRequestOptions {}
 }
 
-export class SeedSimpleApiClient {
-    protected readonly _options: NormalizedClientOptionsWithAuth<SeedSimpleApiClient.Options>;
+export class AcmeSdkClient {
+    protected readonly _options: NormalizedClientOptionsWithAuth<AcmeSdkClient.Options>;
     protected _user: UserClient | undefined;
 
-    constructor(options: SeedSimpleApiClient.Options) {
+    constructor(options: AcmeSdkClient.Options) {
         this._options = normalizeClientOptionsWithAuth(options);
     }
 
diff --git a/src/api/resources/user/client/Client.ts b/src/api/resources/user/client/Client.ts
index 246f78ffd26..444e6746f51 100644
--- a/src/api/resources/user/client/Client.ts
+++ b/src/api/resources/user/client/Client.ts
@@ -6,7 +6,7 @@ import { mergeHeaders } from "../../../../core/headers.js";
 import * as core from "../../../../core/index.js";
 import { handleNonStatusCodeError } from "../../../../errors/handleNonStatusCodeError.js";
 import * as errors from "../../../../errors/index.js";
-import type * as SeedSimpleApi from "../../../index.js";
+import type * as acme from "../../../index.js";
 
 export declare namespace UserClient {
     export type Options = BaseClientOptions;
@@ -28,14 +28,14 @@ export class UserClient {
      * @example
      *     await client.user.get("id")
      */
-    public get(id: string, requestOptions?: UserClient.RequestOptions): core.HttpResponsePromise<SeedSimpleApi.User> {
+    public get(id: string, requestOptions?: UserClient.RequestOptions): core.HttpResponsePromise<acme.User> {
         return core.HttpResponsePromise.fromPromise(this.__get(id, requestOptions));
     }
 
     private async __get(
         id: string,
         requestOptions?: UserClient.RequestOptions,
-    ): Promise<core.WithRawResponse<SeedSimpleApi.User>> {
+    ): Promise<core.WithRawResponse<acme.User>> {
         const _authRequest: core.AuthRequest = await this._options.authProvider.getAuthRequest();
         const _headers: core.Fetcher.Args["headers"] = mergeHeaders(
             _authRequest.headers,
@@ -58,11 +58,11 @@ export class UserClient {
             logging: this._options.logging,
         });
         if (_response.ok) {
-            return { data: _response.body as SeedSimpleApi.User, rawResponse: _response.rawResponse };
+            return { data: _response.body as acme.User, rawResponse: _response.rawResponse };
         }
 
         if (_response.error.reason === "status-code") {
-            throw new errors.SeedSimpleApiError({
+            throw new errors.AcmeApiError({
                 statusCode: _response.error.statusCode,
                 body: _response.error.body,
                 rawResponse: _response.rawResponse,
diff --git a/src/auth/BearerAuthProvider.ts b/src/auth/BearerAuthProvider.ts
index 63b42583d7c..a697a85c92e 100644
--- a/src/auth/BearerAuthProvider.ts
+++ b/src/auth/BearerAuthProvider.ts
@@ -23,7 +23,7 @@ export class BearerAuthProvider implements core.AuthProvider {
     } = {}): Promise<core.AuthRequest> {
         const token = await core.Supplier.get(this.options[TOKEN_PARAM]);
         if (token == null) {
-            throw new errors.SeedSimpleApiError({
+            throw new errors.AcmeApiError({
                 message: BearerAuthProvider.AUTH_CONFIG_ERROR_MESSAGE,
             });
         }
diff --git a/src/environments.ts b/src/environments.ts
index 0955c381185..83f3465b762 100644
--- a/src/environments.ts
+++ b/src/environments.ts
@@ -1,10 +1,8 @@
 // This file was auto-generated by Fern from our API Definition.
 
-export const SeedSimpleApiEnvironment = {
+export const AcmeApiEnvironment = {
     Production: "https://api.example.com",
     Staging: "https://staging-api.example.com",
 } as const;
 
-export type SeedSimpleApiEnvironment =
-    | typeof SeedSimpleApiEnvironment.Production
-    | typeof SeedSimpleApiEnvironment.Staging;
+export type AcmeApiEnvironment = typeof AcmeApiEnvironment.Production | typeof AcmeApiEnvironment.Staging;
diff --git a/src/errors/SeedSimpleApiError.ts b/src/errors/AcmeApiError.ts
similarity index 96%
rename from src/errors/SeedSimpleApiError.ts
rename to src/errors/AcmeApiError.ts
index 36e4d16bb4c..d5aba858b42 100644
--- a/src/errors/SeedSimpleApiError.ts
+++ b/src/errors/AcmeApiError.ts
@@ -3,7 +3,7 @@
 import type * as core from "../core/index.js";
 import { toJson } from "../core/json.js";
 
-export class SeedSimpleApiError extends Error {
+export class AcmeApiError extends Error {
     public readonly statusCode?: number;
     public readonly body?: unknown;
     public readonly rawResponse?: core.RawResponse;
diff --git a/src/errors/SeedSimpleApiTimeoutError.ts b/src/errors/AcmeRequestTimeoutError.ts
similarity index 89%
rename from src/errors/SeedSimpleApiTimeoutError.ts
rename to src/errors/AcmeRequestTimeoutError.ts
index 32c4e3952f3..0a167a41815 100644
--- a/src/errors/SeedSimpleApiTimeoutError.ts
+++ b/src/errors/AcmeRequestTimeoutError.ts
@@ -1,6 +1,6 @@
 // This file was auto-generated by Fern from our API Definition.
 
-export class SeedSimpleApiTimeoutError extends Error {
+export class AcmeRequestTimeoutError extends Error {
     public readonly cause?: unknown;
 
     constructor(message: string, opts?: { cause?: unknown }) {
diff --git a/src/errors/handleNonStatusCodeError.ts b/src/errors/handleNonStatusCodeError.ts
index 378b1282b6c..ba52573855f 100644
--- a/src/errors/handleNonStatusCodeError.ts
+++ b/src/errors/handleNonStatusCodeError.ts
@@ -11,28 +11,28 @@ export function handleNonStatusCodeError(
 ): never {
     switch (error.reason) {
         case "non-json":
-            throw new errors.SeedSimpleApiError({
+            throw new errors.AcmeApiError({
                 statusCode: error.statusCode,
                 body: error.rawBody,
                 rawResponse: rawResponse,
             });
         case "body-is-null":
-            throw new errors.SeedSimpleApiError({
+            throw new errors.AcmeApiError({
                 statusCode: error.statusCode,
                 rawResponse: rawResponse,
             });
         case "timeout":
-            throw new errors.SeedSimpleApiTimeoutError(`Timeout exceeded when calling ${method} ${path}.`, {
+            throw new errors.AcmeRequestTimeoutError(`Timeout exceeded when calling ${method} ${path}.`, {
                 cause: error.cause,
             });
         case "unknown":
-            throw new errors.SeedSimpleApiError({
+            throw new errors.AcmeApiError({
                 message: error.errorMessage,
                 rawResponse: rawResponse,
                 cause: error.cause,
             });
         default:
-            throw new errors.SeedSimpleApiError({
+            throw new errors.AcmeApiError({
                 message: "Unknown error",
                 rawResponse: rawResponse,
             });
diff --git a/src/errors/index.ts b/src/errors/index.ts
index 9045c013be1..f8699785977 100644
--- a/src/errors/index.ts
+++ b/src/errors/index.ts
@@ -1,2 +1,2 @@
-export { SeedSimpleApiError } from "./SeedSimpleApiError.js";
-export { SeedSimpleApiTimeoutError } from "./SeedSimpleApiTimeoutError.js";
+export { AcmeApiError } from "./AcmeApiError.js";
+export { AcmeRequestTimeoutError } from "./AcmeRequestTimeoutError.js";
diff --git a/src/index.ts b/src/index.ts
index 8ffb3e9d711..531fe61c5a7 100644
--- a/src/index.ts
+++ b/src/index.ts
@@ -1,6 +1,6 @@
-export * as SeedSimpleApi from "./api/index.js";
+export * as acme from "./api/index.js";
 export type { BaseClientOptions, BaseRequestOptions } from "./BaseClient.js";
-export { SeedSimpleApiClient } from "./Client.js";
-export { SeedSimpleApiEnvironment } from "./environments.js";
-export { SeedSimpleApiError, SeedSimpleApiTimeoutError } from "./errors/index.js";
+export { AcmeSdkClient } from "./Client.js";
+export { AcmeApiEnvironment } from "./environments.js";
+export { AcmeApiError, AcmeRequestTimeoutError } from "./errors/index.js";
 export * from "./exports.js";
diff --git a/tests/wire/user.test.ts b/tests/wire/user.test.ts
index dc48354f2aa..d57da49bd66 100644
--- a/tests/wire/user.test.ts
+++ b/tests/wire/user.test.ts
@@ -1,12 +1,12 @@
 // This file was auto-generated by Fern from our API Definition.
 
-import { SeedSimpleApiClient } from "../../src/Client";
+import { AcmeSdkClient } from "../../src/Client";
 import { mockServerPool } from "../mock-server/MockServerPool";
 
 describe("UserClient", () => {
     test("get", async () => {
         const server = mockServerPool.createServer();
-        const client = new SeedSimpleApiClient({ maxRetries: 0, token: "test", environment: server.baseUrl });
+        const client = new AcmeSdkClient({ maxRetries: 0, token: "test", environment: server.baseUrl });
 
         const rawResponseBody = { id: "id", name: "name", email: "email" };
 
